diff --git a/src/Entity/ApiToken.php b/src/Entity/ApiToken.php
index 2fbd7c3..65fb949 100644
--- a/src/Entity/ApiToken.php
+++ b/src/Entity/ApiToken.php
@@ -8,6 +8,8 @@ use Doctrine\ORM\Mapping as ORM;
 #[ORM\Entity(repositoryClass: ApiTokenRepository::class)]
 class ApiToken
 {
+    private const PERSONAL_ACCESS_TOKEN_PREFIX = 'tcp_';
+
     #[ORM\Id]
     #[ORM\GeneratedValue]
     #[ORM\Column]
@@ -21,11 +23,16 @@ class ApiToken
     private ?\DateTimeImmutable $expiresAt = null;
 
     #[ORM\Column(length: 68)]
-    private string $token = null;
+    private string $token;
 
     #[ORM\Column]
     private array $scopes = [];
 
+    public function __construct(string $tokenType = self::PERSONAL_ACCESS_TOKEN_PREFIX)
+    {
+        $this->token = $tokenType.bin2hex(random_bytes(32));
+    }
+
     public function getId(): ?int
     {
         return $this->id;
