diff --git a/src/Factory/ApiTokenFactory.php b/src/Factory/ApiTokenFactory.php
index 490d37a..b497ce8 100644
--- a/src/Factory/ApiTokenFactory.php
+++ b/src/Factory/ApiTokenFactory.php
@@ -48,8 +48,10 @@ final class ApiTokenFactory extends ModelFactory
     {
         return [
             'ownedBy' => UserFactory::new(),
-            'scopes' => [],
-            'token' => self::faker()->text(64),
+            'scopes' => [
+                'ROLE_TREASURE_CREATE',
+                'ROLE_USER_EDIT'
+            ],
         ];
     }
 
