diff --git a/src/Controller/SecurityController.php b/src/Controller/SecurityController.php
index e41bdad..c3bf39b 100644
--- a/src/Controller/SecurityController.php
+++ b/src/Controller/SecurityController.php
@@ -2,6 +2,7 @@
 
 namespace App\Controller;
 
+use ApiPlatform\Api\IriConverterInterface;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
@@ -10,7 +11,7 @@ use Symfony\Component\Security\Http\Attribute\CurrentUser;
 class SecurityController extends AbstractController
 {
     #[Route('/login', name: 'app_login', methods: ['POST'])]
-    public function login(#[CurrentUser] $user = null): Response
+    public function login(IriConverterInterface $iriConverter, #[CurrentUser] $user = null): Response
     {
         if (!$user) {
             return $this->json([
@@ -18,8 +19,8 @@ class SecurityController extends AbstractController
             ], 401);
         }
 
-        return $this->json([
-            'user' => $user->getId(),
+        return new Response(null, 204, [
+            'Location' => $iriConverter->getIriFromResource($user),
         ]);
     }
 }
