diff --git a/tests/Functional/DragonTreasureResourceTest.php b/tests/Functional/DragonTreasureResourceTest.php
index 518f98e..43f6827 100644
--- a/tests/Functional/DragonTreasureResourceTest.php
+++ b/tests/Functional/DragonTreasureResourceTest.php
@@ -64,6 +64,9 @@ class DragonTreasureResourceTest extends ApiTestCase
         $this->browser()
             ->post('/api/treasures', [
                 'json' => [],
+                'headers' => [
+                    'Authorization' => 'Bearer FOO'
+                ]
             ])
             ->dump()
             ->assertStatus(422)
