diff --git a/tests/Functional/DragonTreasureResourceTest.php b/tests/Functional/DragonTreasureResourceTest.php
index c4e7545..584bd56 100644
--- a/tests/Functional/DragonTreasureResourceTest.php
+++ b/tests/Functional/DragonTreasureResourceTest.php
@@ -41,16 +41,10 @@ class DragonTreasureResourceTest extends KernelTestCase
 
     public function testPostToCreateTreasure(): void
     {
-        $user = UserFactory::createOne(['password' => 'pass']);
+        $user = UserFactory::createOne();
 
         $this->browser()
-            ->post('/login', [
-                'json' => [
-                    'email' => $user->getEmail(),
-                    'password' => 'pass',
-                ],
-            ])
-            ->assertStatus(204)
+            ->actingAs($user)
             ->post('/api/treasures', [
                 'json' => [],
             ])
