diff --git a/tests/Functional/DragonTreasureResourceTest.php b/tests/Functional/DragonTreasureResourceTest.php
index b61bb9b..300c297 100644
--- a/tests/Functional/DragonTreasureResourceTest.php
+++ b/tests/Functional/DragonTreasureResourceTest.php
@@ -74,7 +74,6 @@ class DragonTreasureResourceTest extends ApiTestCase
                     'Authorization' => 'Bearer '.$token->getToken()
                 ]
             ])
-            ->dump()
             ->assertStatus(422)
         ;
     }
