diff --git a/tests/Functional/DragonTreasureResourceTest.php b/tests/Functional/DragonTreasureResourceTest.php
index 7b75151..86d9a76 100644
--- a/tests/Functional/DragonTreasureResourceTest.php
+++ b/tests/Functional/DragonTreasureResourceTest.php
@@ -2,6 +2,7 @@
 
 namespace App\Tests\Functional;
 
+use App\Factory\ApiTokenFactory;
 use App\Factory\DragonTreasureFactory;
 use App\Factory\UserFactory;
 use Zenstruck\Browser\HttpOptions;
@@ -61,11 +62,15 @@ class DragonTreasureResourceTest extends ApiTestCase
 
     public function testPostToCreateTreasureWithApiKey(): void
     {
+        $token = ApiTokenFactory::createOne([
+            'scopes' => []
+        ]);
+
         $this->browser()
             ->post('/api/treasures', [
                 'json' => [],
                 'headers' => [
-                    'Authorization' => 'Bearer FOO'
+                    'Authorization' => 'Bearer '.$token->getToken()
                 ]
             ])
             ->dump()
