diff --git a/assets/vue/LoginForm.vue b/assets/vue/LoginForm.vue
index 628ac02..ba163fc 100644
--- a/assets/vue/LoginForm.vue
+++ b/assets/vue/LoginForm.vue
@@ -70,7 +70,7 @@ const handleSubmit = async () => {
     const response = await fetch('/login', {
         method: 'POST',
         headers: {
-            'Content-Type': 'application/json'
+            //'Content-Type': 'application/json'
         },
         body: JSON.stringify({
             email: email.value,
