diff --git a/tests/Functional/DragonTreasureResourceTest.php b/tests/Functional/DragonTreasureResourceTest.php
index e3bada7..78a9118 100644
--- a/tests/Functional/DragonTreasureResourceTest.php
+++ b/tests/Functional/DragonTreasureResourceTest.php
@@ -56,7 +56,7 @@ class DragonTreasureResourceTest extends KernelTestCase
                 'value' => 1000,
                 'coolFactor' => 5,
                 'owner' => '/api/users/'.$user->getId(),
-            ]))
+            ])->withHeader('Accept', 'application/ld+json'))
             ->assertStatus(201)
             ->dump()
             ->assertJsonMatches('name', 'A shiny thing')
