diff --git a/tests/Functional/DragonTreasureResourceTest.php b/tests/Functional/DragonTreasureResourceTest.php
index 392ce77..518f98e 100644
--- a/tests/Functional/DragonTreasureResourceTest.php
+++ b/tests/Functional/DragonTreasureResourceTest.php
@@ -58,4 +58,15 @@ class DragonTreasureResourceTest extends ApiTestCase
             ->assertJsonMatches('name', 'A shiny thing')
         ;
     }
+
+    public function testPostToCreateTreasureWithApiKey(): void
+    {
+        $this->browser()
+            ->post('/api/treasures', [
+                'json' => [],
+            ])
+            ->dump()
+            ->assertStatus(422)
+        ;
+    }
 }
