diff --git a/tests/Functional/DragonTreasureResourceTest.php b/tests/Functional/DragonTreasureResourceTest.php
index e9c3764..8e684d9 100644
--- a/tests/Functional/DragonTreasureResourceTest.php
+++ b/tests/Functional/DragonTreasureResourceTest.php
@@ -44,13 +44,7 @@ class DragonTreasureResourceTest extends KernelTestCase
         $user = UserFactory::createOne(['password' => 'pass']);
 
         $this->browser()
-            ->post('/login', [
-                'json' => [
-                    'email' => $user->getEmail(),
-                    'password' => 'pass',
-                ],
-            ])
-            ->assertStatus(204)
+            ->actingAs($user)
             ->post('/api/treasures', [
                 'json' => [],
             ])
