diff --git a/tests/Functional/DragonTreasureResourceTest.php b/tests/Functional/DragonTreasureResourceTest.php
index d9926cc..ec7d1f4 100644
--- a/tests/Functional/DragonTreasureResourceTest.php
+++ b/tests/Functional/DragonTreasureResourceTest.php
@@ -37,4 +37,15 @@ class DragonTreasureResourceTest extends KernelTestCase
             'plunderedAtAgo',
         ]);
     }
+
+    public function testPostToCreateTreasure(): void
+    {
+        $this->browser()
+            ->post('/api/treasures', [
+                'json' => [],
+            ])
+            ->assertStatus(422)
+            ->dump()
+        ;
+    }
 }
