diff --git a/tests/Functional/ApiTestCase.php b/tests/Functional/ApiTestCase.php
new file mode 100644
index 0000000..615cb3c
--- /dev/null
+++ b/tests/Functional/ApiTestCase.php
@@ -0,0 +1,26 @@
+<?php
+
+namespace App\Tests\Functional;
+
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use Zenstruck\Browser\HttpOptions;
+use Zenstruck\Browser\KernelBrowser;
+use Zenstruck\Browser\Test\HasBrowser;
+
+abstract class ApiTestCase extends KernelTestCase
+{
+    use HasBrowser {
+        browser as baseKernelBrowser;
+    }
+
+    protected function browser(): KernelBrowser
+    {
+        return $this->baseKernelBrowser()
+            ->setDefaultHttpOptions(
+                HttpOptions::create()
+                    ->withHeader('Accept', 'application/ld+json')
+
+            )
+        ;
+    }
+}
diff --git a/tests/Functional/DragonTreasureResourceTest.php b/tests/Functional/DragonTreasureResourceTest.php
index 62ae57a..def75fe 100644
--- a/tests/Functional/DragonTreasureResourceTest.php
+++ b/tests/Functional/DragonTreasureResourceTest.php
@@ -4,13 +4,12 @@ namespace App\Tests\Functional;
 
 use App\Factory\DragonTreasureFactory;
 use App\Factory\UserFactory;
-use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
 use Zenstruck\Browser\HttpOptions;
 use Zenstruck\Browser\Json;
 use Zenstruck\Browser\Test\HasBrowser;
 use Zenstruck\Foundry\Test\ResetDatabase;
 
-class DragonTreasureResourceTest extends KernelTestCase
+class DragonTreasureResourceTest extends ApiTestCase
 {
     use HasBrowser;
     use ResetDatabase;
@@ -56,7 +55,7 @@ class DragonTreasureResourceTest extends KernelTestCase
                 'value' => 1000,
                 'coolFactor' => 5,
                 'owner' => '/api/users/'.$user->getId(),
-            ])->withHeader('Accept', 'application/ld+json'))
+            ]))
             ->assertStatus(201)
             ->dump()
             ->assertJsonMatches('name', 'A shiny thing')
