diff --git a/tests/Functional/DragonTreasureResourceTest.php b/tests/Functional/DragonTreasureResourceTest.php
index 17075d4..3016200 100644
--- a/tests/Functional/DragonTreasureResourceTest.php
+++ b/tests/Functional/DragonTreasureResourceTest.php
@@ -55,7 +55,6 @@ class DragonTreasureResourceTest extends ApiTestCase
                 'owner' => '/api/users/'.$user->getId(),
             ]))
             ->assertStatus(201)
-            ->dump()
             ->assertJsonMatches('name', 'A shiny thing')
         ;
     }
