diff --git a/src/Automapper/UserApiToUserMapperConfiguration.php b/src/Automapper/UserApiToUserMapperConfiguration.php
index 9ee3ff6..4ed41ac 100644
--- a/src/Automapper/UserApiToUserMapperConfiguration.php
+++ b/src/Automapper/UserApiToUserMapperConfiguration.php
@@ -23,6 +23,11 @@ class UserApiToUserMapperConfiguration implements MapperConfigurationInterface
         assert($metadata instanceof MapperMetadata);
 
         $metadata->forMember('password', function (UserApi $userApi, User $user) {
+            // if the password is not set, we keep the old one
+            if (!$userApi->password) {
+                return $user->getPassword();
+            }
+
             return $this->userPasswordHasher->hashPassword($user, $userApi->password);
         });
     }
