diff --git a/src/Automapper/UserApiToUserMapperConfiguration.php b/src/Automapper/UserApiToUserMapperConfiguration.php
index 604677e..857b883 100644
--- a/src/Automapper/UserApiToUserMapperConfiguration.php
+++ b/src/Automapper/UserApiToUserMapperConfiguration.php
@@ -11,7 +11,7 @@ class UserApiToUserMapperConfiguration implements MapperConfigurationInterface
 {
     public function process(MapperGeneratorMetadataInterface $metadata): void
     {
-        dd($metadata);
+        dd($metadata->getPropertiesMapping());
     }
 
     public function getSource(): string
