diff --git a/src/Entity/User.php b/src/Entity/User.php
index 070018a..42a1295 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -131,12 +131,16 @@ class User implements UserInterface, PasswordAuthenticatedUserInterface
         return $this->username;
     }
 
-//    public function setUsername(string $username): self
-//    {
-//        $this->username = $username;
-//
-//        return $this;
-//    }
+    public function setUsername(string $username): self
+    {
+        $this->username = $username;
+
+        return $this;
+    }
+
+    public function setFlameThrowingDistance()
+    {
+    }
 
     /**
      * @return Collection<int, DragonTreasure>
