diff --git a/src/State/EntityClassDtoStateProcessor.php b/src/State/EntityClassDtoStateProcessor.php
index 1a47edb..3dd5d69 100644
--- a/src/State/EntityClassDtoStateProcessor.php
+++ b/src/State/EntityClassDtoStateProcessor.php
@@ -32,7 +32,6 @@ class EntityClassDtoStateProcessor implements ProcessorInterface
         assert($data instanceof UserApi);
 
         $entity = $this->mapDtoToEntity($data);
-        dd($entity);
 
         if ($operation instanceof DeleteOperationInterface) {
             $this->removeProcessor->process($entity, $operation, $uriVariables, $context);
