diff --git a/src/State/EntityClassDtoStateProvider.php b/src/State/EntityClassDtoStateProvider.php
index 548532fb..8f15fbfd 100644
--- a/src/State/EntityClassDtoStateProvider.php
+++ b/src/State/EntityClassDtoStateProvider.php
@@ -10,6 +10,7 @@ use ApiPlatform\Metadata\Operation;
 use ApiPlatform\State\Pagination\TraversablePaginator;
 use ApiPlatform\State\ProviderInterface;
 use App\ApiResource\UserApi;
+use App\MicroMapper\AutoMapper;
 use Symfony\Component\DependencyInjection\Attribute\Autowire;
 
 class EntityClassDtoStateProvider implements ProviderInterface
@@ -17,6 +18,7 @@ class EntityClassDtoStateProvider implements ProviderInterface
     public function __construct(
         #[Autowire(service: CollectionProvider::class)] private ProviderInterface $collectionProvider,
         #[Autowire(service: ItemProvider::class)] private ProviderInterface $itemProvider,
+        private AutoMapper $autoMapper
     )
     {
 
@@ -52,13 +54,6 @@ class EntityClassDtoStateProvider implements ProviderInterface
 
     private function mapEntityToDto(object $entity): object
     {
-        $dto = new UserApi();
-        $dto->id = $entity->getId();
-        $dto->email = $entity->getEmail();
-        $dto->username = $entity->getUsername();
-        $dto->dragonTreasures = $entity->getPublishedDragonTreasures()->getValues();
-        $dto->flameThrowingDistance = rand(1, 10);
-
-        return $dto;
+        return $this->autoMapper->map($entity, UserApi::class);
     }
 }
