diff --git a/src/Automapper/UserApiToEntityMapperConfiguration.php b/src/Automapper/UserApiToEntityMapperConfiguration.php
index 3666736..e1b5954 100644
--- a/src/Automapper/UserApiToEntityMapperConfiguration.php
+++ b/src/Automapper/UserApiToEntityMapperConfiguration.php
@@ -23,6 +23,11 @@ class UserApiToEntityMapperConfiguration implements MapperConfigurationInterface
         assert($metadata instanceof MapperMetadata);
 
         $metadata->forMember('password', function (UserApi $userApi, User $user) {
+            // if the password is not set, we keep the old one
+            if (!$userApi->password) {
+                return $user->getPassword();
+            }
+
             return $this->userPasswordHasher->hashPassword($user, $userApi->password);
         });
     }
