diff --git a/src/Automapper/UserApiToEntityMapperConfiguration.php b/src/Automapper/UserApiToEntityMapperConfiguration.php
index 2b12508..72ad045 100644
--- a/src/Automapper/UserApiToEntityMapperConfiguration.php
+++ b/src/Automapper/UserApiToEntityMapperConfiguration.php
@@ -6,12 +6,17 @@ use App\ApiResource\UserApi;
 use App\Entity\User;
 use Jane\Bundle\AutoMapperBundle\Configuration\MapperConfigurationInterface;
 use Jane\Component\AutoMapper\MapperGeneratorMetadataInterface;
+use Jane\Component\AutoMapper\MapperMetadata;
 
 class UserApiToEntityMapperConfiguration implements MapperConfigurationInterface
 {
     public function process(MapperGeneratorMetadataInterface $metadata): void
     {
-        dd($metadata->getPropertiesMapping());
+        assert($metadata instanceof MapperMetadata);
+
+        $metadata->forMember('password', function (UserApi $userApi) {
+            return 'BAAAR';
+        });
     }
 
     public function getSource(): string
diff --git a/src/State/EntityClassDtoStateProcessor.php b/src/State/EntityClassDtoStateProcessor.php
index 3dd5d69..1a47edb 100644
--- a/src/State/EntityClassDtoStateProcessor.php
+++ b/src/State/EntityClassDtoStateProcessor.php
@@ -32,6 +32,7 @@ class EntityClassDtoStateProcessor implements ProcessorInterface
         assert($data instanceof UserApi);
 
         $entity = $this->mapDtoToEntity($data);
+        dd($entity);
 
         if ($operation instanceof DeleteOperationInterface) {
             $this->removeProcessor->process($entity, $operation, $uriVariables, $context);
