diff --git a/tests/Functional/DragonTreasureResourceTest.php b/tests/Functional/DragonTreasureResourceTest.php
index 3069457..1c439b6 100644
--- a/tests/Functional/DragonTreasureResourceTest.php
+++ b/tests/Functional/DragonTreasureResourceTest.php
@@ -54,7 +54,6 @@ class DragonTreasureResourceTest extends ApiTestCase
                 'description' => 'It sparkles when I wave it in the air.',
                 'value' => 1000,
                 'coolFactor' => 5,
-                'owner' => '/api/users/'.$user->getId(),
             ]))
             ->assertStatus(201)
             ->assertJsonMatches('name', 'A shiny thing')
