diff --git a/src/Automapper/UserApiToEntityMapperConfiguration.php b/src/Automapper/UserApiToEntityMapperConfiguration.php
index 6ccd21d..2b12508 100644
--- a/src/Automapper/UserApiToEntityMapperConfiguration.php
+++ b/src/Automapper/UserApiToEntityMapperConfiguration.php
@@ -11,7 +11,7 @@ class UserApiToEntityMapperConfiguration implements MapperConfigurationInterface
 {
     public function process(MapperGeneratorMetadataInterface $metadata): void
     {
-        dd($metadata);
+        dd($metadata->getPropertiesMapping());
     }
 
     public function getSource(): string
