diff --git a/src/Automapper/EntityToUserApiMapperConfiguration.php b/src/Automapper/EntityToUserApiMapperConfiguration.php
index 849cb85..8067e18 100644
--- a/src/Automapper/EntityToUserApiMapperConfiguration.php
+++ b/src/Automapper/EntityToUserApiMapperConfiguration.php
@@ -17,6 +17,8 @@ class EntityToUserApiMapperConfiguration implements MapperConfigurationInterface
 
         $metadata->forMember('dragonTreasures', fn (User $user) => $user->getPublishedDragonTreasures()->getValues());
         $metadata->forMember('flameThrowingDistance', fn (User $user) => rand(0, 10));
+        // always start the plain password with null
+        $metadata->forMember('password', fn (User $user) => null);
     }
 
     public function getSource(): string
