diff --git a/src/Automapper/EntityToUserApiMapperConfiguration.php b/src/Automapper/EntityToUserApiMapperConfiguration.php
new file mode 100644
index 0000000..993e30b
--- /dev/null
+++ b/src/Automapper/EntityToUserApiMapperConfiguration.php
@@ -0,0 +1,31 @@
+<?php
+
+namespace App\Automapper;
+
+use App\ApiResource\UserApi;
+use App\Entity\User;
+use Doctrine\Common\Collections\ArrayCollection;
+use Jane\Bundle\AutoMapperBundle\Configuration\MapperConfigurationInterface;
+use Jane\Component\AutoMapper\MapperGeneratorMetadataInterface;
+use Jane\Component\AutoMapper\MapperMetadata;
+
+class EntityToUserApiMapperConfiguration implements MapperConfigurationInterface
+{
+    public function process(MapperGeneratorMetadataInterface $metadata): void
+    {
+        assert($metadata instanceof MapperMetadata);
+
+        $metadata->forMember('dragonTreasures', fn (User $user) => $user->getPublishedDragonTreasures()->getValues());
+    }
+
+    public function getSource(): string
+    {
+        return User::class;
+    }
+
+    public function getTarget(): string
+    {
+        return UserApi::class;
+    }
+
+}
