diff --git a/src/Automapper/UserApiToUserMapperConfiguration.php b/src/Automapper/UserApiToUserMapperConfiguration.php
index 857b883..03a78ce 100644
--- a/src/Automapper/UserApiToUserMapperConfiguration.php
+++ b/src/Automapper/UserApiToUserMapperConfiguration.php
@@ -6,12 +6,17 @@ use App\ApiResource\UserApi;
 use App\Entity\User;
 use Jane\Bundle\AutoMapperBundle\Configuration\MapperConfigurationInterface;
 use Jane\Component\AutoMapper\MapperGeneratorMetadataInterface;
+use Jane\Component\AutoMapper\MapperMetadata;
 
 class UserApiToUserMapperConfiguration implements MapperConfigurationInterface
 {
     public function process(MapperGeneratorMetadataInterface $metadata): void
     {
-        dd($metadata->getPropertiesMapping());
+        assert($metadata instanceof MapperMetadata);
+
+        $metadata->forMember('password', function (UserApi $userApi) {
+            return 'BAAAR';
+        });
     }
 
     public function getSource(): string
diff --git a/src/State/UserApiStateProcessor.php b/src/State/UserApiStateProcessor.php
index 3ae2eda..270dda5 100644
--- a/src/State/UserApiStateProcessor.php
+++ b/src/State/UserApiStateProcessor.php
@@ -32,6 +32,7 @@ class UserApiStateProcessor implements ProcessorInterface
         assert($data instanceof UserApi);
 
         $user = $this->mapDtoToEntity($data);
+        dd($user);
 
         if ($operation instanceof DeleteOperationInterface) {
             $this->removeProcessor->process($user, $operation, $uriVariables, $context);
