diff --git a/src/Entity/User.php b/src/Entity/User.php
index 53e62a0..cc8317e 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -186,12 +186,12 @@ class User implements UserInterface, PasswordAuthenticatedUserInterface
         return $this->username;
     }
 
-    public function setUsername(string $username): self
-    {
-        $this->username = $username;
-
-        return $this;
-    }
+//    public function setUsername(string $username): self
+//    {
+//        $this->username = $username;
+//
+//        return $this;
+//    }
 
     /**
      * @return Collection<int, DragonTreasure>
