diff --git a/src/Automapper/UserApiToEntityMapper.php b/src/Automapper/UserApiToEntityMapper.php
index ff3e5180..d050c0be 100644
--- a/src/Automapper/UserApiToEntityMapper.php
+++ b/src/Automapper/UserApiToEntityMapper.php
@@ -6,18 +6,33 @@ use App\ApiResource\UserApi;
 use App\Entity\User;
 use App\MicroMapper\AsMapper;
 use App\MicroMapper\MapperInterface;
+use App\Repository\UserRepository;
 
 #[AsMapper(from: UserApi::class, to: User::class)]
 class UserApiToEntityMapper implements MapperInterface
 {
+    public function __construct(
+        private UserRepository $userRepository,
+    )
+    {
+
+    }
+
     public function init(object $from, string $toClass, array $context): object
     {
-        dd($from, $toClass);
-        // TODO: Implement init() method.
+        $dto = $from;
+        assert($dto instanceof UserApi);
+
+        $userEntity = $dto->id ? $this->userRepository->find($dto->id) : new User();
+        if (!$userEntity) {
+            throw new \Exception('User not found');
+        }
+
+        return $userEntity;
     }
 
     public function populate(object $from, object $to, array $context): object
     {
-        // TODO: Implement populate() method.
+        dd($from, $to);
     }
 }
