diff --git a/src/State/EntityClassDtoStateProvider.php b/src/State/EntityClassDtoStateProvider.php
index 86f0ea4..6dae620 100644
--- a/src/State/EntityClassDtoStateProvider.php
+++ b/src/State/EntityClassDtoStateProvider.php
@@ -34,7 +34,6 @@ class EntityClassDtoStateProvider implements ProviderInterface
             foreach ($entities as $entity) {
                 $dtos[] = $this->mapEntityToDto($entity);
             }
-            dd($dtos);
 
             return new TraversablePaginator(
                 new \ArrayIterator($dtos),
