diff --git a/src/Automapper/UserApiToUserMapperConfiguration.php b/src/Automapper/UserApiToUserMapperConfiguration.php
index 03a78ce..9ee3ff6 100644
--- a/src/Automapper/UserApiToUserMapperConfiguration.php
+++ b/src/Automapper/UserApiToUserMapperConfiguration.php
@@ -7,15 +7,23 @@ use App\Entity\User;
 use Jane\Bundle\AutoMapperBundle\Configuration\MapperConfigurationInterface;
 use Jane\Component\AutoMapper\MapperGeneratorMetadataInterface;
 use Jane\Component\AutoMapper\MapperMetadata;
+use Symfony\Component\PasswordHasher\Hasher\UserPasswordHasherInterface;
 
 class UserApiToUserMapperConfiguration implements MapperConfigurationInterface
 {
+    public function __construct(
+        private UserPasswordHasherInterface $userPasswordHasher
+    )
+    {
+
+    }
+
     public function process(MapperGeneratorMetadataInterface $metadata): void
     {
         assert($metadata instanceof MapperMetadata);
 
-        $metadata->forMember('password', function (UserApi $userApi) {
-            return 'BAAAR';
+        $metadata->forMember('password', function (UserApi $userApi, User $user) {
+            return $this->userPasswordHasher->hashPassword($user, $userApi->password);
         });
     }
 
