diff --git a/src/Automapper/EntityToUserApiMapperConfiguration.php b/src/Automapper/EntityToUserApiMapperConfiguration.php
index 993e30b..849cb85 100644
--- a/src/Automapper/EntityToUserApiMapperConfiguration.php
+++ b/src/Automapper/EntityToUserApiMapperConfiguration.php
@@ -16,6 +16,7 @@ class EntityToUserApiMapperConfiguration implements MapperConfigurationInterface
         assert($metadata instanceof MapperMetadata);
 
         $metadata->forMember('dragonTreasures', fn (User $user) => $user->getPublishedDragonTreasures()->getValues());
+        $metadata->forMember('flameThrowingDistance', fn (User $user) => rand(0, 10));
     }
 
     public function getSource(): string
