diff --git a/src/ApiPlatform/OpenApiFactoryDecorator.php b/src/ApiPlatform/OpenApiFactoryDecorator.php
new file mode 100644
index 0000000..90166f4
--- /dev/null
+++ b/src/ApiPlatform/OpenApiFactoryDecorator.php
@@ -0,0 +1,14 @@
+<?php
+
+namespace App\ApiPlatform;
+
+use ApiPlatform\OpenApi\Factory\OpenApiFactoryInterface;
+use ApiPlatform\OpenApi\OpenApi;
+
+class OpenApiFactoryDecorator implements OpenApiFactoryInterface
+{
+    public function __invoke(array $context = []): OpenApi
+    {
+        // TODO: Implement __invoke() method.
+    }
+}
