diff --git a/src/ApiPlatform/OpenApiFactoryDecorator.php b/src/ApiPlatform/OpenApiFactoryDecorator.php
index d55e20e..f6597d5 100644
--- a/src/ApiPlatform/OpenApiFactoryDecorator.php
+++ b/src/ApiPlatform/OpenApiFactoryDecorator.php
@@ -3,6 +3,7 @@
 namespace App\ApiPlatform;
 
 use ApiPlatform\OpenApi\Factory\OpenApiFactoryInterface;
+use ApiPlatform\OpenApi\Model\SecurityScheme;
 use ApiPlatform\OpenApi\OpenApi;
 use Symfony\Component\DependencyInjection\Attribute\AsDecorator;
 
@@ -17,7 +18,11 @@ class OpenApiFactoryDecorator implements OpenApiFactoryInterface
     {
         $openApi = $this->decorated->__invoke($context);
 
-        dump($openApi);
+        $securitySchemes = $openApi->getComponents()->getSecuritySchemes() ?: new \ArrayObject();
+        $securitySchemes['access_token'] = new SecurityScheme(
+            type: 'http',
+            scheme: 'bearer',
+        );
 
         return $openApi;
     }
