diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index 4a62d66..8e23d56 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -6,6 +6,7 @@ use ApiPlatform\Doctrine\Orm\Filter\BooleanFilter;
 use ApiPlatform\Doctrine\Orm\Filter\SearchFilter;
 use ApiPlatform\Doctrine\Orm\Filter\RangeFilter;
 use ApiPlatform\Metadata\ApiFilter;
+use ApiPlatform\Metadata\ApiProperty;
 use ApiPlatform\Metadata\ApiResource;
 use ApiPlatform\Metadata\Delete;
 use ApiPlatform\Metadata\Get;
@@ -126,6 +127,7 @@ class DragonTreasure
     #[ORM\Column]
     #[ApiFilter(BooleanFilter::class)]
     #[Groups(['treasure:read', 'treasure:write'])]
+    #[ApiProperty(readable: false)]
     private bool $isPublished = false;
 
     #[ORM\ManyToOne(inversedBy: 'dragonTreasures')]
