diff --git a/tests/Functional/DragonTreasureResourceTest.php b/tests/Functional/DragonTreasureResourceTest.php
index 94378a6..51e833e 100644
--- a/tests/Functional/DragonTreasureResourceTest.php
+++ b/tests/Functional/DragonTreasureResourceTest.php
@@ -122,7 +122,9 @@ class DragonTreasureResourceTest extends ApiTestCase
     public function testAdminCanPatchToEditTreasure(): void
     {
         $admin = UserFactory::new()->asAdmin()->create();
-        $treasure = DragonTreasureFactory::createOne();
+        $treasure = DragonTreasureFactory::createOne([
+            'isPublished' => false,
+        ]);
 
         $this->browser()
             ->actingAs($admin)
@@ -133,6 +135,7 @@ class DragonTreasureResourceTest extends ApiTestCase
             ])
             ->assertStatus(200)
             ->assertJsonMatches('value', 12345)
+            ->assertJsonMatches('isPublished', false)
         ;
     }
 }
