diff --git a/src/ApiResource/UserApi.php b/src/ApiResource/UserApi.php
index 545aa78c..6c313fe4 100644
--- a/src/ApiResource/UserApi.php
+++ b/src/ApiResource/UserApi.php
@@ -16,6 +16,7 @@ use App\Entity\User;
 #[ApiResource(
     shortName: 'User',
     normalizationContext: [AbstractNormalizer::IGNORED_ATTRIBUTES => ['flameThrowingDistance']],
+    denormalizationContext: [AbstractNormalizer::IGNORED_ATTRIBUTES => ['flameThrowingDistance']],
     paginationItemsPerPage: 5,
     provider: EntityToDtoStateProvider::class,
     processor: EntityClassDtoStateProcessor::class,
