diff --git a/src/ApiResource/UserApi.php b/src/ApiResource/UserApi.php
index 09abfa0..149e8e2 100644
--- a/src/ApiResource/UserApi.php
+++ b/src/ApiResource/UserApi.php
@@ -24,6 +24,7 @@ use App\Entity\User;
 ])]
 class UserApi
 {
+    #[ApiProperty(readable: false, identifier: true)]
     public ?int $id = null;
 
     public ?string $email = null;
diff --git a/tests/Functional/UserResourceTest.php b/tests/Functional/UserResourceTest.php
index 551d5b0..80c016f 100644
--- a/tests/Functional/UserResourceTest.php
+++ b/tests/Functional/UserResourceTest.php
@@ -26,6 +26,7 @@ class UserResourceTest extends ApiTestCase
             ->assertStatus(201)
             ->use(function (Json $json) {
                 $json->assertMissing('password');
+                $json->assertMissing('id');
             })
             ->post('/login', [
                 'json' => [
