diff --git a/src/State/EntityClassDtoStateProcessor.php b/src/State/EntityClassDtoStateProcessor.php
index 1654145..9f3b4fb 100644
--- a/src/State/EntityClassDtoStateProcessor.php
+++ b/src/State/EntityClassDtoStateProcessor.php
@@ -39,6 +39,7 @@ class EntityClassDtoStateProcessor implements ProcessorInterface
 
         $this->persistProcessor->process($entity, $operation, $uriVariables, $context);
         $data->id = $entity->getId();
+        dump($data);
 
         return $data;
     }
diff --git a/tests/Functional/UserResourceTest.php b/tests/Functional/UserResourceTest.php
index 80c016f..80b418b 100644
--- a/tests/Functional/UserResourceTest.php
+++ b/tests/Functional/UserResourceTest.php
@@ -47,9 +47,11 @@ class UserResourceTest extends ApiTestCase
             ->patch('/api/users/' . $user->getId(), [
                 'json' => [
                     'username' => 'changed',
+                    'flameThrowingDistance' => 999,
                 ],
                 'headers' => ['Content-Type' => 'application/merge-patch+json']
             ])
+            ->dump()
             ->assertStatus(200);
     }
 
