diff --git a/src/ApiResource/UserApi.php b/src/ApiResource/UserApi.php
index 149e8e21..ea3b32ea 100644
--- a/src/ApiResource/UserApi.php
+++ b/src/ApiResource/UserApi.php
@@ -24,7 +24,7 @@ use App\Entity\User;
 ])]
 class UserApi
 {
-    #[ApiProperty(readable: false, identifier: true)]
+    #[ApiProperty(readable: false, writable: false, identifier: true)]
     public ?int $id = null;
 
     public ?string $email = null;
