diff --git a/src/ApiResource/UserApi.php b/src/ApiResource/UserApi.php
index 62244a7..73b454b 100644
--- a/src/ApiResource/UserApi.php
+++ b/src/ApiResource/UserApi.php
@@ -16,7 +16,7 @@ use App\State\UserApiStateProvider;
 class UserApi
 {
     #[ApiProperty(readable: false, writable: false, identifier: true)]
-    public ?int $id;
+    public readonly ?int $id;
 
     public ?string $email = null;
 
