diff --git a/src/ApiResource/UserApi.php b/src/ApiResource/UserApi.php
index e3c482d..36a454b 100644
--- a/src/ApiResource/UserApi.php
+++ b/src/ApiResource/UserApi.php
@@ -16,6 +16,7 @@ use App\Entity\User;
 #[ApiResource(
     shortName: 'User',
     normalizationContext: [AbstractNormalizer::IGNORED_ATTRIBUTES => ['tmpProperty']],
+    denormalizationContext: [AbstractNormalizer::IGNORED_ATTRIBUTES => ['tmpProperty']],
     paginationItemsPerPage: 5,
     provider: UserApiStateProvider::class,
     processor: UserApiStateProcessor::class,
