diff --git a/tests/Functional/UserResourceTest.php b/tests/Functional/UserResourceTest.php
index 80c016fa..18d5dd31 100644
--- a/tests/Functional/UserResourceTest.php
+++ b/tests/Functional/UserResourceTest.php
@@ -47,6 +47,7 @@ class UserResourceTest extends ApiTestCase
             ->patch('/api/users/' . $user->getId(), [
                 'json' => [
                     'username' => 'changed',
+                    'id' => 47,
                 ],
                 'headers' => ['Content-Type' => 'application/merge-patch+json']
             ])
