diff --git a/src/ApiResource/UserApi.php b/src/ApiResource/UserApi.php
index bd75847..22979d9 100644
--- a/src/ApiResource/UserApi.php
+++ b/src/ApiResource/UserApi.php
@@ -27,7 +27,7 @@ use Doctrine\Common\Collections\Collection;
 class UserApi
 {
     #[ApiProperty(readable: false, writable: false, identifier: true)]
-    public ?int $id;
+    public readonly ?int $id;
 
     public ?string $email = null;
 
