diff --git a/src/ApiResource/UserApi.php b/src/ApiResource/UserApi.php
index e8d8f69..2611b36 100644
--- a/src/ApiResource/UserApi.php
+++ b/src/ApiResource/UserApi.php
@@ -26,7 +26,7 @@ use Doctrine\Common\Collections\Collection;
 ])]
 class UserApi
 {
-    #[ApiProperty(identifier: true)]
+    #[ApiProperty(readable: false, identifier: true)]
     public ?int $id;
 
     public ?string $email = null;
diff --git a/tests/Functional/UserResourceTest.php b/tests/Functional/UserResourceTest.php
index 551d5b0..80c016f 100644
--- a/tests/Functional/UserResourceTest.php
+++ b/tests/Functional/UserResourceTest.php
@@ -26,6 +26,7 @@ class UserResourceTest extends ApiTestCase
             ->assertStatus(201)
             ->use(function (Json $json) {
                 $json->assertMissing('password');
+                $json->assertMissing('id');
             })
             ->post('/login', [
                 'json' => [
