diff --git a/tests/Functional/UserResourceTest.php b/tests/Functional/UserResourceTest.php
index 120f547..496632a 100644
--- a/tests/Functional/UserResourceTest.php
+++ b/tests/Functional/UserResourceTest.php
@@ -46,6 +46,7 @@ class UserResourceTest extends ApiTestCase
             ->patch('/api/user_apis/' . $user->getId(), [
                 'json' => [
                     'username' => 'changed',
+                    'id' => 47,
                 ],
                 'headers' => ['Content-Type' => 'application/merge-patch+json']
             ])
