diff --git a/src/ApiResource/UserApi.php b/src/ApiResource/UserApi.php
index 2611b36..bd75847 100644
--- a/src/ApiResource/UserApi.php
+++ b/src/ApiResource/UserApi.php
@@ -26,7 +26,7 @@ use Doctrine\Common\Collections\Collection;
 ])]
 class UserApi
 {
-    #[ApiProperty(readable: false, identifier: true)]
+    #[ApiProperty(readable: false, writable: false, identifier: true)]
     public ?int $id;
 
     public ?string $email = null;
