diff --git a/tests/Functional/UserResourceTest.php b/tests/Functional/UserResourceTest.php
index 6fc7615..120f547 100644
--- a/tests/Functional/UserResourceTest.php
+++ b/tests/Functional/UserResourceTest.php
@@ -43,7 +43,7 @@ class UserResourceTest extends ApiTestCase
 
         $this->browser()
             ->actingAs($user)
-            ->patch('/api/users/' . $user->getId(), [
+            ->patch('/api/user_apis/' . $user->getId(), [
                 'json' => [
                     'username' => 'changed',
                 ],
