diff --git a/src/ApiResource/UserApi.php b/src/ApiResource/UserApi.php
index 22979d9..94e16b7 100644
--- a/src/ApiResource/UserApi.php
+++ b/src/ApiResource/UserApi.php
@@ -27,7 +27,7 @@ use Doctrine\Common\Collections\Collection;
 class UserApi
 {
     #[ApiProperty(readable: false, writable: false, identifier: true)]
-    public readonly ?int $id;
+    public int $id;
 
     public ?string $email = null;
 
@@ -46,9 +46,8 @@ class UserApi
 
     public int $flameThrowingDistance = 0;
 
-    public function __construct(int $id = null)
+    public function __construct()
     {
-        $this->id = $id;
         $this->dragonTreasures = new ArrayCollection();
     }
 }
diff --git a/src/State/UserApiStateProvider.php b/src/State/UserApiStateProvider.php
index 56535fd..772dbae 100644
--- a/src/State/UserApiStateProvider.php
+++ b/src/State/UserApiStateProvider.php
@@ -54,7 +54,8 @@ class UserApiStateProvider implements ProviderInterface
 
     private function mapEntityToDto(User $user): UserApi
     {
-        $userApi = new UserApi($user->getId());
+        $userApi = new UserApi();
+        $userApi->id = $user->getId();
         $userApi->email = $user->getEmail();
         $userApi->username = $user->getUsername();
         $userApi->dragonTreasures = new ArrayCollection($user->getPublishedDragonTreasures()->getValues());
