diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index 4f4bd2ef..3e931433 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -137,6 +137,11 @@ class DragonTreasure
     #[ApiFilter(SearchFilter::class, strategy: 'exact')]
     private ?User $owner = null;
 
+    /**
+     * @var bool Non-persisted property to help determine if the treasure is owned by the authenticated user
+     */
+    private bool $isOwnedByAuthenticatedUser;
+
     public function __construct(string $name = null)
     {
         $this->name = $name;
@@ -248,4 +253,14 @@ class DragonTreasure
 
         return $this;
     }
+
+    public function isOwnedByAuthenticatedUser(): bool
+    {
+        return $this->isOwnedByAuthenticatedUser;
+    }
+
+    public function setIsOwnedByAuthenticatedUser(bool $isOwnedByAuthenticatedUser)
+    {
+        $this->isOwnedByAuthenticatedUser = $isOwnedByAuthenticatedUser;
+    }
 }
