diff --git a/src/ApiResource/DailyQuest.php b/src/ApiResource/DailyQuest.php
index 6c7ff581..9348b3cc 100644
--- a/src/ApiResource/DailyQuest.php
+++ b/src/ApiResource/DailyQuest.php
@@ -11,5 +11,10 @@ use App\State\DailyQuestStateProvider;
 )]
 class DailyQuest
 {
+    public int $id;
 
+    public function __construct(int $id)
+    {
+        $this->id = $id;
+    }
 }
diff --git a/src/State/DailyQuestStateProvider.php b/src/State/DailyQuestStateProvider.php
index 6e6b627f..59c83cca 100644
--- a/src/State/DailyQuestStateProvider.php
+++ b/src/State/DailyQuestStateProvider.php
@@ -11,8 +11,8 @@ class DailyQuestStateProvider implements ProviderInterface
     public function provide(Operation $operation, array $uriVariables = [], array $context = []): object|array|null
     {
         return [
-            new DailyQuest(),
-            new DailyQuest(),
+            new DailyQuest(4),
+            new DailyQuest(5),
         ];
     }
 }
