diff --git a/src/ApiResource/DailyQuest.php b/src/ApiResource/DailyQuest.php
index 9e7efa89..8d48a562 100644
--- a/src/ApiResource/DailyQuest.php
+++ b/src/ApiResource/DailyQuest.php
@@ -32,6 +32,7 @@ class DailyQuest
     public int $difficultyLevel;
     public DailyQuestStatusEnum $status;
     public \DateTimeInterface $lastUpdated;
+    public array $treasures = [];
 
     public function __construct(\DateTimeInterface $day)
     {
diff --git a/src/State/DailyQuestStateProvider.php b/src/State/DailyQuestStateProvider.php
index 99cc1822..5891e2fa 100644
--- a/src/State/DailyQuestStateProvider.php
+++ b/src/State/DailyQuestStateProvider.php
@@ -7,9 +7,16 @@ use ApiPlatform\Metadata\Operation;
 use ApiPlatform\State\ProviderInterface;
 use App\ApiResource\DailyQuest;
 use App\Enum\DailyQuestStatusEnum;
+use App\Repository\DragonTreasureRepository;
 
 class DailyQuestStateProvider implements ProviderInterface
 {
+    public function __construct(
+        private DragonTreasureRepository $treasureRepository,
+    )
+    {
+    }
+
     public function provide(Operation $operation, array $uriVariables = [], array $context = []): object|array|null
     {
         if ($operation instanceof CollectionOperationInterface) {
@@ -23,6 +30,8 @@ class DailyQuestStateProvider implements ProviderInterface
 
     private function createQuests(): array
     {
+        $treasures = $this->treasureRepository->findBy([], [], 10);
+
         $quests = [];
         for ($i = 0; $i < 50; $i++) {
             $quest = new DailyQuest(new \DateTimeImmutable(sprintf('- %d days', $i)));
@@ -32,6 +41,10 @@ class DailyQuestStateProvider implements ProviderInterface
             $quest->status = $i % 2 === 0 ? DailyQuestStatusEnum::ACTIVE : DailyQuestStatusEnum::COMPLETED;
             $quest->lastUpdated = new \DateTimeImmutable(sprintf('- %d days', rand(10, 100)));
 
+            $randomTreasuresKeys = array_rand($treasures, rand(1, 3));
+            $randomTreasures = array_map(fn($key) => $treasures[$key], (array) $randomTreasuresKeys);
+            $quest->treasures = $randomTreasures;
+
             $quests[$quest->getDayString()] = $quest;
         }
 
