diff --git a/src/ApiResource/DailyQuest.php b/src/ApiResource/DailyQuest.php
index d76139c..f1bb2a9 100644
--- a/src/ApiResource/DailyQuest.php
+++ b/src/ApiResource/DailyQuest.php
@@ -4,6 +4,7 @@ namespace App\ApiResource;
 
 use ApiPlatform\Metadata\ApiProperty;
 use ApiPlatform\Metadata\ApiResource;
+use App\Enum\DailyQuestStatusEnum;
 use App\State\DailyQuestStateProvider;
 use Symfony\Component\Serializer\Annotation\Ignore;
 
@@ -15,6 +16,10 @@ class DailyQuest
 {
     #[Ignore]
     public \DateTimeInterface $day;
+    public string $questName;
+    public string $description;
+    public int $difficultyLevel;
+    public DailyQuestStatusEnum $status;
 
     public function __construct(\DateTimeInterface $day)
     {
diff --git a/src/Enum/DailyQuestStatusEnum.php b/src/Enum/DailyQuestStatusEnum.php
new file mode 100644
index 0000000..09c7c1a
--- /dev/null
+++ b/src/Enum/DailyQuestStatusEnum.php
@@ -0,0 +1,9 @@
+<?php
+
+namespace App\Enum;
+
+enum DailyQuestStatusEnum: string
+{
+    case ACTIVE = 'active';
+    case COMPLETED = 'completed';
+}
diff --git a/src/State/DailyQuestStateProvider.php b/src/State/DailyQuestStateProvider.php
index 5a76b05..6828f94 100644
--- a/src/State/DailyQuestStateProvider.php
+++ b/src/State/DailyQuestStateProvider.php
@@ -5,14 +5,28 @@ namespace App\State;
 use ApiPlatform\Metadata\Operation;
 use ApiPlatform\State\ProviderInterface;
 use App\ApiResource\DailyQuest;
+use App\Enum\DailyQuestStatusEnum;
 
 class DailyQuestStateProvider implements ProviderInterface
 {
     public function provide(Operation $operation, array $uriVariables = [], array $context = []): object|array|null
     {
-        return [
-            new DailyQuest(new \DateTime('now')),
-            new DailyQuest(new \DateTime('yesterday')),
-        ];
+        return $this->createQuests();
+    }
+
+    private function createQuests(): array
+    {
+        $quests = [];
+        for ($i = 0; $i < 50; $i++) {
+            $quest = new DailyQuest(new \DateTimeImmutable(sprintf('- %d days', $i)));
+            $quest->questName = sprintf('Quest %d', $i);
+            $quest->description = sprintf('Description %d', $i);
+            $quest->difficultyLevel = $i % 10;
+            $quest->status = $i % 2 === 0 ? DailyQuestStatusEnum::ACTIVE : DailyQuestStatusEnum::COMPLETED;
+
+            $quests[$quest->getDayString()] = $quest;
+        }
+
+        return $quests;
     }
 }
