diff --git a/tests/Functional/DailyQuestResourceTest.php b/tests/Functional/DailyQuestResourceTest.php
index a192c957..122c6fa8 100644
--- a/tests/Functional/DailyQuestResourceTest.php
+++ b/tests/Functional/DailyQuestResourceTest.php
@@ -2,6 +2,7 @@
 
 namespace App\Tests\Functional;
 
+use App\Factory\DragonTreasureFactory;
 use Zenstruck\Foundry\Test\Factories;
 use Zenstruck\Foundry\Test\ResetDatabase;
 
@@ -12,6 +13,9 @@ class DailyQuestResourceTest extends ApiTestCase
 
     public function testPatchCanUpdateStatus()
     {
+        // quests need at least some treasures to be available
+        DragonTreasureFactory::createMany(5);
+
         $day = new \DateTime('-2 days');
         $this->browser()
             ->patch('/api/quests/'.$day->format('Y-m-d'), [
