diff --git a/src/ApiResource/DailyQuest.php b/src/ApiResource/DailyQuest.php
index f4a58dd..d17ab36 100644
--- a/src/ApiResource/DailyQuest.php
+++ b/src/ApiResource/DailyQuest.php
@@ -8,6 +8,7 @@ use ApiPlatform\Metadata\Get;
 use ApiPlatform\Metadata\GetCollection;
 use ApiPlatform\Metadata\Patch;
 use App\Enum\DailyQuestStatusEnum;
+use App\State\DailyQuestStateProcessor;
 use App\State\DailyQuestStateProvider;
 use Symfony\Component\Serializer\Annotation\Ignore;
 
@@ -16,9 +17,11 @@ use Symfony\Component\Serializer\Annotation\Ignore;
     operations: [
         new GetCollection(),
         new Get(),
-        new Patch(),
+        new Patch(
+            processor: DailyQuestStateProcessor::class,
+        ),
     ],
-    provider: DailyQuestStateProvider::class
+    provider: DailyQuestStateProvider::class,
 )]
 class DailyQuest
 {
diff --git a/src/State/DailyQuestStateProcessor.php b/src/State/DailyQuestStateProcessor.php
index 41f00ce..cee09b5 100644
--- a/src/State/DailyQuestStateProcessor.php
+++ b/src/State/DailyQuestStateProcessor.php
@@ -9,6 +9,6 @@ class DailyQuestStateProcessor implements ProcessorInterface
 {
     public function process(mixed $data, Operation $operation, array $uriVariables = [], array $context = []): void
     {
-        // Handle the state
+        dd($data);
     }
 }
