diff --git a/src/ApiResource/DailyQuest.php b/src/ApiResource/DailyQuest.php
index f1bb2a94..f4a58dd8 100644
--- a/src/ApiResource/DailyQuest.php
+++ b/src/ApiResource/DailyQuest.php
@@ -4,13 +4,21 @@ namespace App\ApiResource;
 
 use ApiPlatform\Metadata\ApiProperty;
 use ApiPlatform\Metadata\ApiResource;
+use ApiPlatform\Metadata\Get;
+use ApiPlatform\Metadata\GetCollection;
+use ApiPlatform\Metadata\Patch;
 use App\Enum\DailyQuestStatusEnum;
 use App\State\DailyQuestStateProvider;
 use Symfony\Component\Serializer\Annotation\Ignore;
 
 #[ApiResource(
     shortName: 'Quest',
-    provider: DailyQuestStateProvider::class,
+    operations: [
+        new GetCollection(),
+        new Get(),
+        new Patch(),
+    ],
+    provider: DailyQuestStateProvider::class
 )]
 class DailyQuest
 {
