diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index 18a46dc5..2ea31f2e 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -254,6 +254,8 @@ class DragonTreasure
         return $this;
     }
 
+    #[Groups(['treasure:read'])]
+    #[SerializedName('isMine')]
     public function isOwnedByAuthenticatedUser(): bool
     {
         if (!isset($this->isOwnedByAuthenticatedUser)) {
