diff --git a/src/ApiResource/DailyQuest.php b/src/ApiResource/DailyQuest.php
index 9348b3cc..3ef41b4e 100644
--- a/src/ApiResource/DailyQuest.php
+++ b/src/ApiResource/DailyQuest.php
@@ -11,10 +11,10 @@ use App\State\DailyQuestStateProvider;
 )]
 class DailyQuest
 {
-    public int $id;
+    public \DateTimeInterface $day;
 
-    public function __construct(int $id)
+    public function __construct(\DateTimeInterface $day)
     {
-        $this->id = $id;
+        $this->day = $day;
     }
 }
diff --git a/src/State/DailyQuestStateProvider.php b/src/State/DailyQuestStateProvider.php
index 59c83cca..5a76b05c 100644
--- a/src/State/DailyQuestStateProvider.php
+++ b/src/State/DailyQuestStateProvider.php
@@ -11,8 +11,8 @@ class DailyQuestStateProvider implements ProviderInterface
     public function provide(Operation $operation, array $uriVariables = [], array $context = []): object|array|null
     {
         return [
-            new DailyQuest(4),
-            new DailyQuest(5),
+            new DailyQuest(new \DateTime('now')),
+            new DailyQuest(new \DateTime('yesterday')),
         ];
     }
 }
