diff --git a/tests/Functional/DailyQuestResourceTest.php b/tests/Functional/DailyQuestResourceTest.php
index 52742f54..8fe9254a 100644
--- a/tests/Functional/DailyQuestResourceTest.php
+++ b/tests/Functional/DailyQuestResourceTest.php
@@ -21,7 +21,6 @@ class DailyQuestResourceTest extends ApiTestCase
                 'headers' => ['Content-Type' => 'application/merge-patch+json']
             ])
             ->assertStatus(200)
-            ->dump()
             ->assertJsonMatches('status', 'completed')
         ;
     }
