diff --git a/src/ApiResource/DailyQuest.php b/src/ApiResource/DailyQuest.php
index 8d48a56..46ad254 100644
--- a/src/ApiResource/DailyQuest.php
+++ b/src/ApiResource/DailyQuest.php
@@ -7,6 +7,7 @@ use ApiPlatform\Metadata\ApiResource;
 use ApiPlatform\Metadata\Get;
 use ApiPlatform\Metadata\GetCollection;
 use ApiPlatform\Metadata\Patch;
+use App\Entity\DragonTreasure;
 use App\Enum\DailyQuestStatusEnum;
 use App\State\DailyQuestStateProcessor;
 use App\State\DailyQuestStateProvider;
@@ -32,6 +33,9 @@ class DailyQuest
     public int $difficultyLevel;
     public DailyQuestStatusEnum $status;
     public \DateTimeInterface $lastUpdated;
+    /**
+     * @var DragonTreasure[]
+     */
     public array $treasures = [];
 
     public function __construct(\DateTimeInterface $day)
