diff --git a/src/ApiResource/DailyQuest.php b/src/ApiResource/DailyQuest.php
index e82e98f..c4b01da 100644
--- a/src/ApiResource/DailyQuest.php
+++ b/src/ApiResource/DailyQuest.php
@@ -12,11 +12,16 @@ use App\State\DailyQuestStateProvider;
 )]
 class DailyQuest
 {
-    #[ApiProperty(identifier: true)]
     public \DateTimeInterface $day;
 
     public function __construct(\DateTimeInterface $day)
     {
         $this->day = $day;
     }
+
+    #[ApiProperty(identifier: true)]
+    public function getDayString(): string
+    {
+        return $this->day->format('Y-m-d');
+    }
 }
