diff --git a/src/ApiResource/DailyQuest.php b/src/ApiResource/DailyQuest.php
index 89cd196b..9e7efa89 100644
--- a/src/ApiResource/DailyQuest.php
+++ b/src/ApiResource/DailyQuest.php
@@ -31,6 +31,7 @@ class DailyQuest
     public string $description;
     public int $difficultyLevel;
     public DailyQuestStatusEnum $status;
+    public \DateTimeInterface $lastUpdated;
 
     public function __construct(\DateTimeInterface $day)
     {
diff --git a/src/State/DailyQuestStateProcessor.php b/src/State/DailyQuestStateProcessor.php
index cee09b5b..e232848e 100644
--- a/src/State/DailyQuestStateProcessor.php
+++ b/src/State/DailyQuestStateProcessor.php
@@ -4,11 +4,14 @@ namespace App\State;
 
 use ApiPlatform\Metadata\Operation;
 use ApiPlatform\State\ProcessorInterface;
+use App\ApiResource\DailyQuest;
 
 class DailyQuestStateProcessor implements ProcessorInterface
 {
     public function process(mixed $data, Operation $operation, array $uriVariables = [], array $context = []): void
     {
-        dd($data);
+        assert($data instanceof DailyQuest);
+
+        $data->lastUpdated = new \DateTimeImmutable('now');
     }
 }
diff --git a/src/State/DailyQuestStateProvider.php b/src/State/DailyQuestStateProvider.php
index 95e36be6..99cc1822 100644
--- a/src/State/DailyQuestStateProvider.php
+++ b/src/State/DailyQuestStateProvider.php
@@ -30,6 +30,7 @@ class DailyQuestStateProvider implements ProviderInterface
             $quest->description = sprintf('Description %d', $i);
             $quest->difficultyLevel = $i % 10;
             $quest->status = $i % 2 === 0 ? DailyQuestStatusEnum::ACTIVE : DailyQuestStatusEnum::COMPLETED;
+            $quest->lastUpdated = new \DateTimeImmutable(sprintf('- %d days', rand(10, 100)));
 
             $quests[$quest->getDayString()] = $quest;
         }
