diff --git a/tests/Functional/UserResourceTest.php b/tests/Functional/UserResourceTest.php
index 869e0cc..0a08b55 100644
--- a/tests/Functional/UserResourceTest.php
+++ b/tests/Functional/UserResourceTest.php
@@ -15,13 +15,14 @@ class UserResourceTest extends ApiTestCase
     public function testPostToCreateUser(): void
     {
         $this->browser()
-            ->post('/api/users', [
+            ->post('/api/user_apis', [
                 'json' => [
                     'email' => 'draggin_in_the_morning@coffee.com',
                     'username' => 'draggin_in_the_morning',
                     'password' => 'password',
                 ]
             ])
+            ->dump()
             ->assertStatus(201)
             ->post('/login', [
                 'json' => [
