diff --git a/src/ApiResource/UserApi.php b/src/ApiResource/UserApi.php
index 16f124f..3ad671e 100644
--- a/src/ApiResource/UserApi.php
+++ b/src/ApiResource/UserApi.php
@@ -7,6 +7,7 @@ use ApiPlatform\Doctrine\Orm\State\Options;
 use ApiPlatform\Metadata\ApiFilter;
 use ApiPlatform\Metadata\ApiResource;
 use App\Entity\DragonTreasure;
+use App\State\EntityClassDtoStateProcessor;
 use App\State\EntityClassDtoStateProvider;
 use App\Entity\User;
 
@@ -14,6 +15,7 @@ use App\Entity\User;
     shortName: 'User',
     paginationItemsPerPage: 5,
     provider: EntityClassDtoStateProvider::class,
+    processor: EntityClassDtoStateProcessor::class,
     stateOptions: new Options(entityClass: User::class),
 )]
 #[ApiFilter(SearchFilter::class, properties: [
diff --git a/src/State/EntityClassDtoStateProcessor.php b/src/State/EntityClassDtoStateProcessor.php
new file mode 100644
index 0000000..b516a9c
--- /dev/null
+++ b/src/State/EntityClassDtoStateProcessor.php
@@ -0,0 +1,14 @@
+<?php
+
+namespace App\State;
+
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProcessorInterface;
+
+class EntityClassDtoStateProcessor implements ProcessorInterface
+{
+    public function process(mixed $data, Operation $operation, array $uriVariables = [], array $context = []): void
+    {
+        // Handle the state
+    }
+}
