diff --git a/src/ApiResource/UserApi.php b/src/ApiResource/UserApi.php
index 4e057f2..8e7d92f 100644
--- a/src/ApiResource/UserApi.php
+++ b/src/ApiResource/UserApi.php
@@ -23,7 +23,7 @@ use Doctrine\Common\Collections\Collection;
 ])]
 class UserApi
 {
-    public ?int $id = null;
+    public ?int $id;
 
     public ?string $email = null;
 
@@ -34,8 +34,9 @@ class UserApi
      */
     public Collection $dragonTreasures;
 
-    public function __construct()
+    public function __construct(int $id = null)
     {
+        $this->id = $id;
         $this->dragonTreasures = new ArrayCollection();
     }
 }
diff --git a/src/State/UserApiStateProvider.php b/src/State/UserApiStateProvider.php
index 5151a49..59f27dc 100644
--- a/src/State/UserApiStateProvider.php
+++ b/src/State/UserApiStateProvider.php
@@ -10,6 +10,6 @@ class UserApiStateProvider implements ProviderInterface
 {
     public function provide(Operation $operation, array $uriVariables = [], array $context = []): object|array|null
     {
-        return [new UserApi()];
+        return [new UserApi(5)];
     }
 }
