diff --git a/src/ApiResource/UserApi.php b/src/ApiResource/UserApi.php
index 9b2f5f3e..4f93e9bc 100644
--- a/src/ApiResource/UserApi.php
+++ b/src/ApiResource/UserApi.php
@@ -7,11 +7,13 @@ use ApiPlatform\Doctrine\Orm\State\Options;
 use ApiPlatform\Metadata\ApiFilter;
 use ApiPlatform\Metadata\ApiResource;
 use App\Entity\DragonTreasure;
+use App\State\EntityToDtoStateProvider;
 use App\Entity\User;
 
 #[ApiResource(
     shortName: 'User',
     paginationItemsPerPage: 5,
+    provider: EntityToDtoStateProvider::class,
     stateOptions: new Options(entityClass: User::class),
 )]
 #[ApiFilter(SearchFilter::class, properties: [
