diff --git a/src/State/UserApiStateProcessor.php b/src/State/UserApiStateProcessor.php
index 4412a00..9eb37df 100644
--- a/src/State/UserApiStateProcessor.php
+++ b/src/State/UserApiStateProcessor.php
@@ -28,6 +28,8 @@ class UserApiStateProcessor implements ProcessorInterface
         $this->entityManager->persist($user);
         $this->entityManager->flush();
 
+        $data->id = $user->getId();
+
         return $data;
     }
 
