diff --git a/src/State/EntityToDtoStateProvider.php b/src/State/EntityToDtoStateProvider.php
new file mode 100644
index 00000000..f014dde5
--- /dev/null
+++ b/src/State/EntityToDtoStateProvider.php
@@ -0,0 +1,14 @@
+<?php
+
+namespace App\State;
+
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProviderInterface;
+
+class EntityToDtoStateProvider implements ProviderInterface
+{
+    public function provide(Operation $operation, array $uriVariables = [], array $context = []): object|array|null
+    {
+        // Retrieve the state from somewhere
+    }
+}
