diff --git a/src/State/EntityClassDtoStateProcessor.php b/src/State/EntityClassDtoStateProcessor.php
index afeca1c..73386bd 100644
--- a/src/State/EntityClassDtoStateProcessor.php
+++ b/src/State/EntityClassDtoStateProcessor.php
@@ -2,27 +2,33 @@
 
 namespace App\State;
 
+use ApiPlatform\Doctrine\Common\State\PersistProcessor;
 use ApiPlatform\Metadata\Operation;
 use ApiPlatform\State\ProcessorInterface;
 use App\ApiResource\UserApi;
 use App\Entity\User;
 use App\Repository\UserRepository;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
 
 class EntityClassDtoStateProcessor implements ProcessorInterface
 {
     public function __construct(
-        private UserRepository $userRepository
+        private UserRepository $userRepository,
+        #[Autowire(service: PersistProcessor::class)] private ProcessorInterface $persistProcessor,
     )
     {
 
     }
 
-    public function process(mixed $data, Operation $operation, array $uriVariables = [], array $context = []): void
+    public function process(mixed $data, Operation $operation, array $uriVariables = [], array $context = [])
     {
         assert($data instanceof UserApi);
 
         $entity = $this->mapDtoToEntity($data);
-        dd($entity);
+
+        $this->persistProcessor->process($entity, $operation, $uriVariables, $context);
+
+        return $data;
     }
 
     private function mapDtoToEntity(object $dto): object
