diff --git a/src/ApiResource/UserApi.php b/src/ApiResource/UserApi.php
index c560cff..fa01e2e 100644
--- a/src/ApiResource/UserApi.php
+++ b/src/ApiResource/UserApi.php
@@ -8,6 +8,7 @@ use ApiPlatform\Doctrine\Orm\State\Options;
 use ApiPlatform\Metadata\ApiFilter;
 use ApiPlatform\Metadata\ApiResource;
 use App\Entity\DragonTreasure;
+use App\State\UserApiStateProcessor;
 use App\State\UserApiStateProvider;
 use App\Entity\User;
 
@@ -15,6 +16,7 @@ use App\Entity\User;
     shortName: 'User',
     paginationItemsPerPage: 5,
     provider: UserApiStateProvider::class,
+    processor: UserApiStateProcessor::class,
     stateOptions: new Options(entityClass: User::class),
 )]
 #[ApiFilter(SearchFilter::class, properties: [
diff --git a/src/State/UserApiStateProcessor.php b/src/State/UserApiStateProcessor.php
new file mode 100644
index 0000000..e44bfd1
--- /dev/null
+++ b/src/State/UserApiStateProcessor.php
@@ -0,0 +1,14 @@
+<?php
+
+namespace App\State;
+
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProcessorInterface;
+
+class UserApiStateProcessor implements ProcessorInterface
+{
+    public function process(mixed $data, Operation $operation, array $uriVariables = [], array $context = []): void
+    {
+        // Handle the state
+    }
+}
