diff --git a/src/ApiResource/UserApi.php b/src/ApiResource/UserApi.php
index 0864ad2..0db392f 100644
--- a/src/ApiResource/UserApi.php
+++ b/src/ApiResource/UserApi.php
@@ -2,8 +2,6 @@
 
 namespace App\ApiResource;
 
-use ApiPlatform\Doctrine\Orm\Filter\SearchFilter;
-use ApiPlatform\Metadata\ApiFilter;
 use ApiPlatform\Metadata\ApiResource;
 use App\Entity\DragonTreasure;
 use Doctrine\Common\Collections\ArrayCollection;
@@ -12,9 +10,6 @@ use Doctrine\Common\Collections\Collection;
 #[ApiResource(
     paginationItemsPerPage: 5,
 )]
-#[ApiFilter(SearchFilter::class, properties: [
-    'username' => 'partial',
-])]
 class UserApi
 {
     public ?int $id = null;
