diff --git a/src/Entity/User.php b/src/Entity/User.php
index 9f9d7a6..4d3efbb 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -44,7 +44,7 @@ class User implements UserInterface, PasswordAuthenticatedUserInterface
     private ?string $password = null;
 
     #[ORM\Column(length: 255, unique: true)]
-    #[Groups(['user:read', 'user:write', 'treasure:item:get'])]
+    #[Groups(['user:read', 'user:write', 'treasure:item:get', 'treasure:write'])]
     #[Assert\NotBlank]
     private ?string $username = null;
 
