diff --git a/src/Entity/User.php b/src/Entity/User.php
index d14f1b7..7c46b5d 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -48,7 +48,7 @@ class User implements UserInterface, PasswordAuthenticatedUserInterface
     #[Assert\NotBlank]
     private ?string $username = null;
 
-    #[ORM\OneToMany(mappedBy: 'owner', targetEntity: DragonTreasure::class)]
+    #[ORM\OneToMany(mappedBy: 'owner', targetEntity: DragonTreasure::class, cascade: ['persist'])]
     #[Groups(['user:read', 'user:write'])]
     private Collection $dragonTreasures;
 
