diff --git a/src/ApiResource/DailyQuest.php b/src/ApiResource/DailyQuest.php
index 31f1c2b..d913c2c 100644
--- a/src/ApiResource/DailyQuest.php
+++ b/src/ApiResource/DailyQuest.php
@@ -33,11 +33,8 @@ class DailyQuest
     public int $difficultyLevel;
     public DailyQuestStatusEnum $status;
     public \DateTimeInterface $lastUpdated;
-    /**
-     * @var QuestTreasure[]
-     */
     #[ApiProperty(genId: false)]
-    public array $treasures = [];
+    public QuestTreasure $treasure;
 
     public function __construct(\DateTimeInterface $day)
     {
diff --git a/src/State/DailyQuestStateProvider.php b/src/State/DailyQuestStateProvider.php
index a833b30..36ad5ca 100644
--- a/src/State/DailyQuestStateProvider.php
+++ b/src/State/DailyQuestStateProvider.php
@@ -42,17 +42,12 @@ class DailyQuestStateProvider implements ProviderInterface
             $quest->status = $i % 2 === 0 ? DailyQuestStatusEnum::ACTIVE : DailyQuestStatusEnum::COMPLETED;
             $quest->lastUpdated = new \DateTimeImmutable(sprintf('- %d days', rand(10, 100)));
 
-            $randomTreasuresKeys = array_rand($treasures, rand(1, 3));
-            $randomTreasures = array_map(fn($key) => $treasures[$key], (array) $randomTreasuresKeys);
-            $questTreasures = [];
-            foreach ($randomTreasures as $treasure) {
-                $questTreasures[] = new QuestTreasure(
-                    $treasure->getName(),
-                    $treasure->getValue(),
-                    $treasure->getCoolFactor(),
-                );
-            }
-            $quest->treasures = $questTreasures;
+            $randomTreasure = $treasures[array_rand($treasures)];
+            $quest->treasure = new QuestTreasure(
+                $randomTreasure->getName(),
+                $randomTreasure->getValue(),
+                $randomTreasure->getCoolFactor(),
+            );
 
             $quests[$quest->getDayString()] = $quest;
         }
