diff --git a/src/Validator/TreasuresAllowedOwnerChangeValidator.php b/src/Validator/TreasuresAllowedOwnerChangeValidator.php
index b2a48c6..9975f5e 100644
--- a/src/Validator/TreasuresAllowedOwnerChangeValidator.php
+++ b/src/Validator/TreasuresAllowedOwnerChangeValidator.php
@@ -30,12 +30,16 @@ class TreasuresAllowedOwnerChangeValidator extends ConstraintValidator
             assert($dragonTreasure instanceof DragonTreasure);
 
             $originalData = $unitOfWork->getOriginalEntityData($dragonTreasure);
-            dd($dragonTreasure, $originalData);
-        }
+            $originalOwnerId = $originalData['owner_id'];
+            $newOwnerId = $dragonTreasure->getOwner()->getId();
+
+            if (!$originalOwnerId || $originalOwnerId === $newOwnerId) {
+                return;
+            }
 
-        // TODO: implement the validation here
-        $this->context->buildViolation($constraint->message)
-            ->setParameter('{{ value }}', $value)
-            ->addViolation();
+            // the owner is being changed
+            $this->context->buildViolation($constraint->message)
+                ->addViolation();
+        }
     }
 }
