diff --git a/src/ApiResource/DailyQuest.php b/src/ApiResource/DailyQuest.php
index 46ad254..3c1f68b 100644
--- a/src/ApiResource/DailyQuest.php
+++ b/src/ApiResource/DailyQuest.php
@@ -34,7 +34,7 @@ class DailyQuest
     public DailyQuestStatusEnum $status;
     public \DateTimeInterface $lastUpdated;
     /**
-     * @var DragonTreasure[]
+     * @var QuestTreasure[]
      */
     public array $treasures = [];
 
diff --git a/src/ApiResource/QuestTreasure.php b/src/ApiResource/QuestTreasure.php
new file mode 100644
index 0000000..08806c2
--- /dev/null
+++ b/src/ApiResource/QuestTreasure.php
@@ -0,0 +1,14 @@
+<?php
+
+namespace App\ApiResource;
+
+class QuestTreasure
+{
+    public function __construct(
+        public string $name,
+        public int $value,
+        public int $coolFactor,
+    )
+    {
+    }
+}
diff --git a/src/State/DailyQuestStateProvider.php b/src/State/DailyQuestStateProvider.php
index 5891e2f..a833b30 100644
--- a/src/State/DailyQuestStateProvider.php
+++ b/src/State/DailyQuestStateProvider.php
@@ -6,6 +6,7 @@ use ApiPlatform\Metadata\CollectionOperationInterface;
 use ApiPlatform\Metadata\Operation;
 use ApiPlatform\State\ProviderInterface;
 use App\ApiResource\DailyQuest;
+use App\ApiResource\QuestTreasure;
 use App\Enum\DailyQuestStatusEnum;
 use App\Repository\DragonTreasureRepository;
 
@@ -43,7 +44,15 @@ class DailyQuestStateProvider implements ProviderInterface
 
             $randomTreasuresKeys = array_rand($treasures, rand(1, 3));
             $randomTreasures = array_map(fn($key) => $treasures[$key], (array) $randomTreasuresKeys);
-            $quest->treasures = $randomTreasures;
+            $questTreasures = [];
+            foreach ($randomTreasures as $treasure) {
+                $questTreasures[] = new QuestTreasure(
+                    $treasure->getName(),
+                    $treasure->getValue(),
+                    $treasure->getCoolFactor(),
+                );
+            }
+            $quest->treasures = $questTreasures;
 
             $quests[$quest->getDayString()] = $quest;
         }
