diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index b127d69..45a6ecc 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -2,6 +2,8 @@
 
 namespace App\Entity;
 
+use ApiPlatform\Doctrine\Orm\Filter\BooleanFilter;
+use ApiPlatform\Metadata\ApiFilter;
 use ApiPlatform\Metadata\ApiResource;
 use ApiPlatform\Metadata\Get;
 use ApiPlatform\Metadata\GetCollection;
@@ -34,6 +36,7 @@ use Symfony\Component\Serializer\Annotation\SerializedName;
     ],
     paginationItemsPerPage: 10
 )]
+#[ApiFilter(BooleanFilter::class, properties: ['isPublished'])]
 class DragonTreasure
 {
     #[ORM\Id]
