diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index 45a6ecc..44ead2f 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -36,7 +36,6 @@ use Symfony\Component\Serializer\Annotation\SerializedName;
     ],
     paginationItemsPerPage: 10
 )]
-#[ApiFilter(BooleanFilter::class, properties: ['isPublished'])]
 class DragonTreasure
 {
     #[ORM\Id]
@@ -67,6 +66,7 @@ class DragonTreasure
     private \DateTimeImmutable $plunderedAt;
 
     #[ORM\Column]
+    #[ApiFilter(BooleanFilter::class)]
     private bool $isPublished = false;
 
     public function __construct(string $name = null)
