diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index 75e2846..a9603a8 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -2,11 +2,13 @@
 
 namespace App\Entity;
 
+use ApiPlatform\Metadata\ApiResource;
 use App\Repository\DragonTreasureRepository;
 use Doctrine\DBAL\Types\Types;
 use Doctrine\ORM\Mapping as ORM;
 
 #[ORM\Entity(repositoryClass: DragonTreasureRepository::class)]
+#[ApiResource]
 class DragonTreasure
 {
     #[ORM\Id]
