diff --git a/src/Factory/DragonTreasureFactory.php b/src/Factory/DragonTreasureFactory.php
index f3871bd..428b5a2 100644
--- a/src/Factory/DragonTreasureFactory.php
+++ b/src/Factory/DragonTreasureFactory.php
@@ -29,6 +29,8 @@ use Zenstruck\Foundry\RepositoryProxy;
  */
 final class DragonTreasureFactory extends ModelFactory
 {
+    private const TREASURE_NAMES = ['pile of gold coins', 'diamond-encrusted throne', 'rare magic staff', 'enchanted sword', 'set of intricately crafted goblets', 'collection of ancient tomes', 'hoard of shiny gemstones', 'chest filled with priceless works of art', 'giant pearl', 'crown made of pure platinum', 'giant egg (possibly a dragon egg?)', 'set of ornate armor', 'set of golden utensils', 'statue carved from a single block of marble', 'collection of rare, antique weapons', 'box of rare, exotic chocolates', 'set of ornate jewelry', 'set of rare, antique books', 'giant ball of yarn', 'life-sized statue of the dragon itself', 'collection of old, used toothbrushes', 'box of mismatched socks', 'set of outdated electronics (such as CRT TVs or floppy disks)', 'giant jar of pickles', 'collection of novelty mugs with silly sayings', 'pile of old board games', 'giant slinky', 'collection of rare, exotic hats'];
+
     /**
      * @see https://symfony.com/bundles/ZenstruckFoundryBundle/current/index.html#factories-as-services
      *
@@ -41,18 +43,16 @@ final class DragonTreasureFactory extends ModelFactory
 
     /**
      * @see https://symfony.com/bundles/ZenstruckFoundryBundle/current/index.html#model-factories
-     *
-     * @todo add your default values here
      */
     protected function getDefaults(): array
     {
         return [
-            'coolFactor' => self::faker()->randomNumber(),
-            'description' => self::faker()->text(),
+            'coolFactor' => self::faker()->numberBetween(1, 10),
+            'description' => self::faker()->paragraph(),
             'isPublished' => self::faker()->boolean(),
-            'name' => self::faker()->text(255),
-            'plunderedAt' => \DateTimeImmutable::createFromMutable(self::faker()->dateTime()),
-            'value' => self::faker()->randomNumber(),
+            'name' => self::faker()->randomElement(self::TREASURE_NAMES),
+            'plunderedAt' => \DateTimeImmutable::createFromMutable(self::faker()->dateTimeBetween('-1 year')),
+            'value' => self::faker()->numberBetween(1000, 1000000),
         ];
     }
 
