diff --git a/README.md b/README.md
index 32d3cba..5290870 100644
--- a/README.md
+++ b/README.md
@@ -42,6 +42,7 @@ Next, build the database and the schema with:
 # but its aware of your database container
 symfony console doctrine:database:create --if-not-exists
 symfony console doctrine:migrations:migrate
+symfony console doctrine:fixtures:load
 ```
 
 If you do *not* want to use Docker, just make sure to start your own
diff --git a/src/DataFixtures/AppFixtures.php b/src/DataFixtures/AppFixtures.php
index 987f6fe..7137f2b 100644
--- a/src/DataFixtures/AppFixtures.php
+++ b/src/DataFixtures/AppFixtures.php
@@ -2,6 +2,7 @@
 
 namespace App\DataFixtures;
 
+use App\Factory\DragonTreasureFactory;
 use Doctrine\Bundle\FixturesBundle\Fixture;
 use Doctrine\Persistence\ObjectManager;
 
@@ -9,9 +10,6 @@ class AppFixtures extends Fixture
 {
     public function load(ObjectManager $manager): void
     {
-        // $product = new Product();
-        // $manager->persist($product);
-
-        $manager->flush();
+        DragonTreasureFactory::createMany(40);
     }
 }
