diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index 4e3d112..88d443b 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -38,7 +38,14 @@ use function Symfony\Component\String\u;
     denormalizationContext: [
         'groups' => ['treasure:write'],
     ],
-    paginationItemsPerPage: 10
+    paginationItemsPerPage: 10,
+    formats: [
+        'jsonld',
+        'json',
+        'html',
+        'jsonhal',
+        'csv' => 'text/csv',
+    ],
 )]
 #[ApiFilter(PropertyFilter::class)]
 class DragonTreasure
