diff --git a/src/State/EntityClassDtoStateProcessor.php b/src/State/EntityClassDtoStateProcessor.php
index dd0cf505..700b1eb7 100644
--- a/src/State/EntityClassDtoStateProcessor.php
+++ b/src/State/EntityClassDtoStateProcessor.php
@@ -4,19 +4,19 @@ namespace App\State;
 
 use ApiPlatform\Doctrine\Common\State\PersistProcessor;
 use ApiPlatform\Doctrine\Common\State\RemoveProcessor;
+use ApiPlatform\Doctrine\Orm\State\Options;
 use ApiPlatform\Metadata\DeleteOperationInterface;
 use ApiPlatform\Metadata\Operation;
 use ApiPlatform\State\ProcessorInterface;
 use App\ApiResource\UserApi;
-use App\Entity\User;
+use Doctrine\ORM\EntityManagerInterface;
 use App\MicroMapper\AutoMapper;
-use App\Repository\UserRepository;
 use Symfony\Component\DependencyInjection\Attribute\Autowire;
 
 class EntityClassDtoStateProcessor implements ProcessorInterface
 {
     public function __construct(
-        private UserRepository $userRepository,
+        private EntityManagerInterface $entityManager,
         #[Autowire(service: PersistProcessor::class)] private ProcessorInterface $persistProcessor,
         #[Autowire(service: RemoveProcessor::class)] private ProcessorInterface $removeProcessor,
         private AutoMapper $autoMapper
@@ -27,11 +27,13 @@ class EntityClassDtoStateProcessor implements ProcessorInterface
 
     public function process(mixed $data, Operation $operation, array $uriVariables = [], array $context = [])
     {
-        dd($operation);
+        $stateOptions = $operation->getStateOptions();
+        assert($stateOptions instanceof Options);
+        $entityClass = $stateOptions->getEntityClass();
 
         assert($data instanceof UserApi);
 
-        $entity = $this->mapDtoToEntity($data);
+        $entity = $this->mapDtoToEntity($data, $entityClass);
 
         if ($operation instanceof DeleteOperationInterface) {
             $this->removeProcessor->process($entity, $operation, $uriVariables, $context);
@@ -45,8 +47,8 @@ class EntityClassDtoStateProcessor implements ProcessorInterface
         return $data;
     }
 
-    private function mapDtoToEntity(object $dto): object
+    private function mapDtoToEntity(object $dto, string $entityClass): object
     {
-        return $this->autoMapper->map($dto, User::class);
+        return $this->autoMapper->map($dto, $entityClass);
     }
 }
