diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index 3261148..013bbe1 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -62,7 +62,7 @@ class DragonTreasure
     private \DateTimeImmutable $plunderedAt;
 
     #[ORM\Column]
-    private ?bool $isPublished = null;
+    private bool $isPublished = false;
 
     public function __construct()
     {
@@ -151,7 +151,7 @@ class DragonTreasure
         return Carbon::instance($this->plunderedAt)->diffForHumans();
     }
 
-    public function getIsPublished(): ?bool
+    public function getIsPublished(): bool
     {
         return $this->isPublished;
     }
