diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index c441b39..46bee49 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -23,6 +23,9 @@ use Doctrine\ORM\Mapping as ORM;
         new Post(),
         new Put(),
         new Patch(),
+    ],
+    normalizationContext: [
+        'groups' => ['treasure:read'],
     ]
 )]
 class DragonTreasure
