diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index 46bee49..33ec6ac 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -12,6 +12,7 @@ use App\Repository\DragonTreasureRepository;
 use Carbon\Carbon;
 use Doctrine\DBAL\Types\Types;
 use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Serializer\Annotation\Groups;
 
 #[ORM\Entity(repositoryClass: DragonTreasureRepository::class)]
 #[ApiResource(
@@ -36,18 +37,22 @@ class DragonTreasure
     private ?int $id = null;
 
     #[ORM\Column(length: 255)]
+    #[Groups(['treasure:read'])]
     private ?string $name = null;
 
     #[ORM\Column(type: Types::TEXT)]
+    #[Groups(['treasure:read'])]
     private ?string $description = null;
 
     /**
      * The estimated value of this treasure, in gold coins.
      */
     #[ORM\Column]
+    #[Groups(['treasure:read'])]
     private ?int $value = null;
 
     #[ORM\Column]
+    #[Groups(['treasure:read'])]
     private ?int $coolFactor = null;
 
     #[ORM\Column]
