diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index 33ec6ac..07d900f 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -27,6 +27,9 @@ use Symfony\Component\Serializer\Annotation\Groups;
     ],
     normalizationContext: [
         'groups' => ['treasure:read'],
+    ],
+    denormalizationContext: [
+        'groups' => ['treasure:write'],
     ]
 )]
 class DragonTreasure
