diff --git a/tests/Functional/DragonTreasureResourceTest.php b/tests/Functional/DragonTreasureResourceTest.php
index 524e465..b10ccdb 100644
--- a/tests/Functional/DragonTreasureResourceTest.php
+++ b/tests/Functional/DragonTreasureResourceTest.php
@@ -154,7 +154,10 @@ class DragonTreasureResourceTest extends ApiTestCase
     public function testPatchUnpublishedWorks()
     {
         $user = UserFactory::createOne();
-        $treasure = DragonTreasureFactory::createOne(['owner' => $user]);
+        $treasure = DragonTreasureFactory::createOne([
+            'owner' => $user,
+            'isPublished' => false,
+        ]);
 
         $this->browser()
             ->actingAs($user)
