diff --git a/src/Controller/SecurityController.php b/src/Controller/SecurityController.php
index 1f4d9a9..eb8e04e 100644
--- a/src/Controller/SecurityController.php
+++ b/src/Controller/SecurityController.php
@@ -5,12 +5,15 @@ namespace App\Controller;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Http\Attribute\CurrentUser;
 
 class SecurityController extends AbstractController
 {
     #[Route('/login', name: 'app_login', methods: ['POST'])]
-    public function login(): Response
+    public function login(#[CurrentUser] $user = null): Response
     {
-        
+        return $this->json([
+            'user' => $user ? $user->getId() : null,
+        ]);
     }
 }
