diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index 4a0beac..86367e6 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -62,6 +62,9 @@ use function Symfony\Component\String\u;
         'groups' => ['treasure:write'],
     ],
     paginationItemsPerPage: 10,
+    extraProperties: [
+        'standard_put' => true,
+    ],
 )]
 #[ApiResource(
     uriTemplate: '/users/{user_id}/treasures.{_format}',
@@ -76,6 +79,9 @@ use function Symfony\Component\String\u;
     normalizationContext: [
         'groups' => ['treasure:read'],
     ],
+    extraProperties: [
+        'standard_put' => true,
+    ],
 )]
 #[ApiFilter(PropertyFilter::class)]
 #[ApiFilter(SearchFilter::class, properties: [
diff --git a/src/Entity/User.php b/src/Entity/User.php
index e536566..ab9ddcd 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -42,6 +42,9 @@ use Symfony\Component\Validator\Constraints as Assert;
     normalizationContext: ['groups' => ['user:read']],
     denormalizationContext: ['groups' => ['user:write']],
     security: 'is_granted("ROLE_USER")',
+    extraProperties: [
+        'standard_put' => true,
+    ],
 )]
 #[ApiResource(
     uriTemplate: '/treasures/{treasure_id}/owner.{_format}',
@@ -54,6 +57,9 @@ use Symfony\Component\Validator\Constraints as Assert;
     ],
     normalizationContext: ['groups' => ['user:read']],
     security: 'is_granted("ROLE_USER")',
+    extraProperties: [
+        'standard_put' => true,
+    ],
 )]
 #[ApiFilter(PropertyFilter::class)]
 #[UniqueEntity(fields: ['email'], message: 'There is already an account with this email')]
