diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index a9603a8..66a5549 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -22,6 +22,9 @@ class DragonTreasure
     #[ORM\Column(type: Types::TEXT)]
     private ?string $description = null;
 
+    /**
+     * The estimated value of this treasure, in gold coins.
+     */
     #[ORM\Column]
     private ?int $value = null;
 
