diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index d1b4d0b..47ab9ec 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -17,8 +17,8 @@ use Doctrine\ORM\Mapping as ORM;
     shortName: 'Treasure',
     description: 'A rare and valuable treasure.',
     operations: [
-        new Get(uriTemplate: '/dragon-plunder/{id}'),
-        new GetCollection(uriTemplate: '/dragon-plunder'),
+        new Get(),
+        new GetCollection(),
         new Post(),
         new Put(),
         new Patch(),
