diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index 69def9f..47ab9ec 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -14,6 +14,7 @@ use Doctrine\ORM\Mapping as ORM;
 
 #[ORM\Entity(repositoryClass: DragonTreasureRepository::class)]
 #[ApiResource(
+    shortName: 'Treasure',
     description: 'A rare and valuable treasure.',
     operations: [
         new Get(),
