diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index ab473de..a922f7e 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -126,7 +126,7 @@ class DragonTreasure
 
     #[ORM\Column]
     #[ApiFilter(BooleanFilter::class)]
-    #[Groups(['admin:read', 'admin:write'])]
+    #[Groups(['admin:read', 'admin:write', 'owner:read'])]
     private bool $isPublished = false;
 
     #[ORM\ManyToOne(inversedBy: 'dragonTreasures')]
