diff --git a/src/Normalizer/AddOwnerGroupsNormalizer.php b/src/Normalizer/AddOwnerGroupsNormalizer.php
index 98ac570..1020171 100644
--- a/src/Normalizer/AddOwnerGroupsNormalizer.php
+++ b/src/Normalizer/AddOwnerGroupsNormalizer.php
@@ -5,7 +5,7 @@ namespace App\Normalizer;
 use Symfony\Component\DependencyInjection\Attribute\AsDecorator;
 use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
 
-#[AsDecorator('serializer')]
+#[AsDecorator('api_platform.jsonld.normalizer.item')]
 class AddOwnerGroupsNormalizer implements NormalizerInterface
 {
     public function __construct(private NormalizerInterface $decorated)
