diff --git a/src/State/DailyQuestStateProvider.php b/src/State/DailyQuestStateProvider.php
index 48336b3..49b3782 100644
--- a/src/State/DailyQuestStateProvider.php
+++ b/src/State/DailyQuestStateProvider.php
@@ -39,7 +39,7 @@ class DailyQuestStateProvider implements ProviderInterface
             );
         }
 
-        $quests = $this->createQuests();
+        $quests = $this->createQuests(0, $this->countTotalQuests());
 
         return $quests[$uriVariables['dayString']] ?? null;
     }
