diff --git a/src/State/DailyQuestStateProvider.php b/src/State/DailyQuestStateProvider.php
index a41ba419..179567ef 100644
--- a/src/State/DailyQuestStateProvider.php
+++ b/src/State/DailyQuestStateProvider.php
@@ -28,9 +28,8 @@ class DailyQuestStateProvider implements ProviderInterface
             $itemsPerPage = $this->pagination->getLimit($operation, $context);
             $offset = $this->pagination->getOffset($operation, $context);
             $totalItems = $this->countTotalQuests();
-            dd($currentPage, $itemsPerPage, $offset, $totalItems);
 
-            $quests = $this->createQuests();
+            $quests = $this->createQuests($offset, $itemsPerPage);
 
             return new TraversablePaginator(
                 new \ArrayIterator($quests),
@@ -45,13 +44,13 @@ class DailyQuestStateProvider implements ProviderInterface
         return $quests[$uriVariables['dayString']] ?? null;
     }
 
-    private function createQuests(): array
+    private function createQuests(int $offset = 0, int $limit = 10): array
     {
         $treasures = $this->treasureRepository->findBy([], [], 10);
         $totalQuests = $this->countTotalQuests();
 
         $quests = [];
-        for ($i = 0; $i < $totalQuests; $i++) {
+        for ($i = $offset; $i < ($offset + $limit) && $i < $totalQuests; $i++) {
             $quest = new DailyQuest(new \DateTimeImmutable(sprintf('- %d days', $i)));
             $quest->questName = sprintf('Quest %d', $i);
             $quest->description = sprintf('Description %d', $i);
