diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index 0c64f71..b127d69 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -31,7 +31,8 @@ use Symfony\Component\Serializer\Annotation\SerializedName;
     ],
     denormalizationContext: [
         'groups' => ['treasure:write'],
-    ]
+    ],
+    paginationItemsPerPage: 10
 )]
 class DragonTreasure
 {
