diff --git a/tests/Functional/UserResourceTest.php b/tests/Functional/UserResourceTest.php
new file mode 100644
index 0000000..a4729cc
--- /dev/null
+++ b/tests/Functional/UserResourceTest.php
@@ -0,0 +1,15 @@
+<?php
+
+namespace App\Tests\Functional;
+
+use Zenstruck\Foundry\Test\ResetDatabase;
+
+class UserResourceTest extends ApiTestCase
+{
+    use ResetDatabase;
+
+    public function testPostToCreateUser(): void
+    {
+
+    }
+}
