diff --git a/src/Entity/User.php b/src/Entity/User.php
index 3127b93..0a5b40b 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -86,6 +86,8 @@ class User implements UserInterface, PasswordAuthenticatedUserInterface
     #[Groups(['user:write'])]
     private ?string $password = null;
 
+    private ?string $plainPassword = null;
+
     #[ORM\Column(length: 255, unique: true)]
     #[Groups(['user:read', 'user:write', 'treasure:item:get', 'treasure:write'])]
     #[Assert\NotBlank]
@@ -270,4 +272,16 @@ class User implements UserInterface, PasswordAuthenticatedUserInterface
     {
         $this->accessTokenScopes = $scopes;
     }
+
+    public function setPlainPassword(string $plainPassword): User
+    {
+        $this->plainPassword = $plainPassword;
+
+        return $this;
+    }
+
+    public function getPlainPassword(): ?string
+    {
+        return $this->plainPassword;
+    }
 }
