diff --git a/src/Entity/User.php b/src/Entity/User.php
index ab9ddcd..fa3696d 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -90,6 +90,8 @@ class User implements UserInterface, PasswordAuthenticatedUserInterface
     #[Groups(['user:write'])]
     private ?string $password = null;
 
+    private ?string $plainPassword = null;
+
     #[ORM\Column(length: 255, unique: true)]
     #[Groups(['user:read', 'user:write', 'treasure:item:get', 'treasure:write'])]
     #[Assert\NotBlank]
@@ -274,4 +276,16 @@ class User implements UserInterface, PasswordAuthenticatedUserInterface
     {
         $this->accessTokenScopes = $scopes;
     }
+
+    public function setPlainPassword(string $plainPassword): User
+    {
+        $this->plainPassword = $plainPassword;
+
+        return $this;
+    }
+
+    public function getPlainPassword(): ?string
+    {
+        return $this->plainPassword;
+    }
 }
