diff --git a/tests/Functional/UserResourceTest.php b/tests/Functional/UserResourceTest.php
index a4729cc..db91c38 100644
--- a/tests/Functional/UserResourceTest.php
+++ b/tests/Functional/UserResourceTest.php
@@ -10,6 +10,22 @@ class UserResourceTest extends ApiTestCase
 
     public function testPostToCreateUser(): void
     {
-
+        $this->browser()
+            ->post('/api/users', [
+                'json' => [
+                    'email' => 'draggin_in_the_morning@coffee.com',
+                    'username' => 'draggin_in_the_morning',
+                    'password' => 'password',
+                ]
+            ])
+            ->assertStatus(201)
+            ->post('/login', [
+                'json' => [
+                    'email' => 'draggin_in_the_morning@coffee.com',
+                    'password' => 'password',
+                ]
+            ])
+            ->assertSuccessful()
+        ;
     }
 }
