diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index d8a2c9e..9506f6a 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -39,7 +39,7 @@ use function Symfony\Component\String\u;
             security: 'is_granted("ROLE_USER")',
         ),
         new Patch(
-            security: 'is_granted("ROLE_USER")',
+            security: 'is_granted("ROLE_USER") and object.owner == user',
         ),
         new Delete(
             security: 'is_granted("ROLE_ADMIN")',
