diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index 440d36a..d8a2c9e 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -38,9 +38,6 @@ use function Symfony\Component\String\u;
         new Post(
             security: 'is_granted("ROLE_USER")',
         ),
-        new Put(
-            security: 'is_granted("ROLE_USER")',
-        ),
         new Patch(
             security: 'is_granted("ROLE_USER")',
         ),
