diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index ed78896..531c424 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -39,7 +39,7 @@ use function Symfony\Component\String\u;
             security: 'is_granted("ROLE_TREASURE_CREATE")',
         ),
         new Patch(
-            security: 'is_granted("ROLE_TREASURE_EDIT")',
+            security: 'is_granted("ROLE_TREASURE_EDIT") and object.owner == user',
         ),
         new Delete(
             security: 'is_granted("ROLE_ADMIN")',
