diff --git a/src/State/EntityToDtoStateProvider.php b/src/State/EntityToDtoStateProvider.php
index 7d9753b..35fc24b 100644
--- a/src/State/EntityToDtoStateProvider.php
+++ b/src/State/EntityToDtoStateProvider.php
@@ -56,7 +56,7 @@ class EntityToDtoStateProvider implements ProviderInterface
         $dto->id = $entity->getId();
         $dto->email = $entity->getEmail();
         $dto->username = $entity->getUsername();
-        $dto->dragonTreasures = $entity->getDragonTreasures()->toArray();
+        $dto->dragonTreasures = $entity->getPublishedDragonTreasures()->getValues();
         $dto->flameThrowingDistance = rand(1, 10);
 
         return $dto;
