diff --git a/src/State/UserApiStateProvider.php b/src/State/UserApiStateProvider.php
index 59f27dc..8416444 100644
--- a/src/State/UserApiStateProvider.php
+++ b/src/State/UserApiStateProvider.php
@@ -2,14 +2,23 @@
 
 namespace App\State;
 
+use ApiPlatform\Doctrine\Orm\State\CollectionProvider;
 use ApiPlatform\Metadata\Operation;
 use ApiPlatform\State\ProviderInterface;
-use App\ApiResource\UserApi;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
 
 class UserApiStateProvider implements ProviderInterface
 {
+    public function __construct(
+        #[Autowire(service: CollectionProvider::class)] private ProviderInterface $collectionProvider
+    )
+    {
+
+    }
+
     public function provide(Operation $operation, array $uriVariables = [], array $context = []): object|array|null
     {
-        return [new UserApi(5)];
+        $users = $this->collectionProvider->provide($operation, $uriVariables, $context);
+        dd($users);
     }
 }
