diff --git a/src/ApiResource/UserApi.php b/src/ApiResource/UserApi.php
index 16f124f8..1d4af8a8 100644
--- a/src/ApiResource/UserApi.php
+++ b/src/ApiResource/UserApi.php
@@ -32,5 +32,5 @@ class UserApi
      */
     public array $dragonTreasures = [];
 
-    public int $flameThrowingDistance = 0;
+//    public int $flameThrowingDistance = 0;
 }
diff --git a/src/State/EntityClassDtoStateProvider.php b/src/State/EntityClassDtoStateProvider.php
index 5f5871b0..36bfe690 100644
--- a/src/State/EntityClassDtoStateProvider.php
+++ b/src/State/EntityClassDtoStateProvider.php
@@ -21,6 +21,8 @@ class EntityClassDtoStateProvider implements ProviderInterface
     {
         $entities = $this->collectionProvider->provide($operation, $uriVariables, $context);
 
+        return $entities;
+
         $dtos = [];
         foreach ($entities as $entity) {
             $dtos[] = $this->mapEntityToDto($entity);
