diff --git a/src/State/EntityToDtoStateProvider.php b/src/State/EntityToDtoStateProvider.php
index 5f7f61e0..7d9753b6 100644
--- a/src/State/EntityToDtoStateProvider.php
+++ b/src/State/EntityToDtoStateProvider.php
@@ -2,6 +2,7 @@
 
 namespace App\State;
 
+use ApiPlatform\Doctrine\Orm\State\ItemProvider;
 use ApiPlatform\Metadata\CollectionOperationInterface;
 use ApiPlatform\Doctrine\Orm\Paginator;
 use ApiPlatform\Doctrine\Orm\State\CollectionProvider;
@@ -14,7 +15,8 @@ use Symfony\Component\DependencyInjection\Attribute\Autowire;
 class EntityToDtoStateProvider implements ProviderInterface
 {
     public function __construct(
-        #[Autowire(service: CollectionProvider::class)] private ProviderInterface $collectionProvider
+        #[Autowire(service: CollectionProvider::class)] private ProviderInterface $collectionProvider,
+        #[Autowire(service: ItemProvider::class)] private ProviderInterface $itemProvider,
     )
     {
 
@@ -39,7 +41,13 @@ class EntityToDtoStateProvider implements ProviderInterface
             );
         }
 
-        dd($uriVariables);
+        $entity = $this->itemProvider->provide($operation, $uriVariables, $context);
+
+        if (!$entity) {
+            return null;
+        }
+
+        return $this->mapEntityToDto($entity);
     }
 
     private function mapEntityToDto(object $entity): object
