diff --git a/src/ApiResource/UserApi.php b/src/ApiResource/UserApi.php
index 4f93e9b..a4f442e 100644
--- a/src/ApiResource/UserApi.php
+++ b/src/ApiResource/UserApi.php
@@ -31,4 +31,6 @@ class UserApi
      * @var array<int, DragonTreasure>
      */
     public array $dragonTreasures = [];
+
+    public int $flameThrowingDistance = 0;
 }
diff --git a/src/State/EntityToDtoStateProvider.php b/src/State/EntityToDtoStateProvider.php
index d7671b7..2fcbaeb 100644
--- a/src/State/EntityToDtoStateProvider.php
+++ b/src/State/EntityToDtoStateProvider.php
@@ -36,6 +36,7 @@ class EntityToDtoStateProvider implements ProviderInterface
         $dto->email = $entity->getEmail();
         $dto->username = $entity->getUsername();
         $dto->dragonTreasures = $entity->getDragonTreasures()->toArray();
+        $dto->flameThrowingDistance = rand(1, 10);
 
         return $dto;
     }
