diff --git a/src/State/EntityClassDtoStateProvider.php b/src/State/EntityClassDtoStateProvider.php
index e059b3a..4a7381f 100644
--- a/src/State/EntityClassDtoStateProvider.php
+++ b/src/State/EntityClassDtoStateProvider.php
@@ -5,6 +5,7 @@ namespace App\State;
 use ApiPlatform\Doctrine\Orm\State\CollectionProvider;
 use ApiPlatform\Metadata\Operation;
 use ApiPlatform\State\ProviderInterface;
+use App\ApiResource\UserApi;
 use Symfony\Component\DependencyInjection\Attribute\Autowire;
 
 class EntityClassDtoStateProvider implements ProviderInterface
@@ -19,6 +20,23 @@ class EntityClassDtoStateProvider implements ProviderInterface
     public function provide(Operation $operation, array $uriVariables = [], array $context = []): object|array|null
     {
         $entities = $this->collectionProvider->provide($operation, $uriVariables, $context);
-        dd(iterator_to_array($entities));
+
+        $dtos = [];
+        foreach ($entities as $entity) {
+            $dtos[] = $this->mapEntityToDto($entity);
+        }
+
+        return $dtos;
+    }
+
+    private function mapEntityToDto(object $entity): object
+    {
+        $dto = new UserApi();
+        $dto->id = $entity->getId();
+        $dto->email = $entity->getEmail();
+        $dto->username = $entity->getUsername();
+        $dto->dragonTreasures = $entity->getDragonTreasures()->toArray();
+
+        return $dto;
     }
 }
