diff --git a/src/State/UserApiStateProvider.php b/src/State/UserApiStateProvider.php
index bfb20d8..2519e78 100644
--- a/src/State/UserApiStateProvider.php
+++ b/src/State/UserApiStateProvider.php
@@ -5,6 +5,8 @@ namespace App\State;
 use ApiPlatform\Doctrine\Orm\State\CollectionProvider;
 use ApiPlatform\Metadata\Operation;
 use ApiPlatform\State\ProviderInterface;
+use App\ApiResource\UserApi;
+use App\Entity\User;
 use Symfony\Component\DependencyInjection\Attribute\Autowire;
 
 class UserApiStateProvider implements ProviderInterface
@@ -19,6 +21,22 @@ class UserApiStateProvider implements ProviderInterface
     public function provide(Operation $operation, array $uriVariables = [], array $context = []): object|array|null
     {
         $users = $this->collectionProvider->provide($operation, $uriVariables, $context);
-        dd(iterator_to_array($users));
+
+        $userDtos = [];
+        foreach ($users as $user) {
+            $userDtos[] = $this->mapEntityToDto($user);
+        }
+
+        return $userDtos;
+    }
+
+    private function mapEntityToDto(User $user): UserApi
+    {
+        $userApi = new UserApi($user->getId());
+        $userApi->email = $user->getEmail();
+        $userApi->username = $user->getUsername();
+        $userApi->dragonTreasures = $user->getDragonTreasures();
+
+        return $userApi;
     }
 }
