diff --git a/src/State/EntityToDtoStateProvider.php b/src/State/EntityToDtoStateProvider.php
index f014dde..51df606 100644
--- a/src/State/EntityToDtoStateProvider.php
+++ b/src/State/EntityToDtoStateProvider.php
@@ -2,13 +2,23 @@
 
 namespace App\State;
 
+use ApiPlatform\Doctrine\Orm\State\CollectionProvider;
 use ApiPlatform\Metadata\Operation;
 use ApiPlatform\State\ProviderInterface;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
 
 class EntityToDtoStateProvider implements ProviderInterface
 {
+    public function __construct(
+        #[Autowire(service: CollectionProvider::class)] private ProviderInterface $collectionProvider
+    )
+    {
+
+    }
+
     public function provide(Operation $operation, array $uriVariables = [], array $context = []): object|array|null
     {
-        // Retrieve the state from somewhere
+        $entities = $this->collectionProvider->provide($operation, $uriVariables, $context);
+        dd($entities);
     }
 }
