diff --git a/src/State/UserApiStateProvider.php b/src/State/UserApiStateProvider.php
index 8e1fc53..c158860 100644
--- a/src/State/UserApiStateProvider.php
+++ b/src/State/UserApiStateProvider.php
@@ -56,7 +56,7 @@ class UserApiStateProvider implements ProviderInterface
         $userApi = new UserApi($user->getId());
         $userApi->email = $user->getEmail();
         $userApi->username = $user->getUsername();
-        $userApi->dragonTreasures = $user->getDragonTreasures();
+        $userApi->dragonTreasures = $user->getPublishedDragonTreasures();
         $userApi->flameThrowingDistance = rand(1, 10);
 
         return $userApi;
