diff --git a/src/State/UserApiStateProvider.php b/src/State/UserApiStateProvider.php
index ba2432f..8d8421f 100644
--- a/src/State/UserApiStateProvider.php
+++ b/src/State/UserApiStateProvider.php
@@ -40,7 +40,7 @@ class UserApiStateProvider implements ProviderInterface
         $userApi = new UserApi($user->getId());
         $userApi->email = $user->getEmail();
         $userApi->username = $user->getUsername();
-        $userApi->dragonTreasures = $user->getDragonTreasures();
+        $userApi->dragonTreasures = $user->getPublishedDragonTreasures();
 
         return $userApi;
     }
