diff --git a/src/State/UserApiStateProvider.php b/src/State/UserApiStateProvider.php
index 35b64a9..ba2432f 100644
--- a/src/State/UserApiStateProvider.php
+++ b/src/State/UserApiStateProvider.php
@@ -26,7 +26,13 @@ class UserApiStateProvider implements ProviderInterface
             return array_map([$this, 'mapEntityToDto'], $users);
         }
 
-        dd($uriVariables);
+        $user = $this->userRepository->find($uriVariables['id']);
+
+        if (!$user) {
+            return null;
+        }
+
+        return $this->mapEntityToDto($user);
     }
 
     private function mapEntityToDto(User $user): UserApi
