diff --git a/src/State/EntityToDtoStateProvider.php b/src/State/EntityToDtoStateProvider.php
index 51df6067..5e8a883f 100644
--- a/src/State/EntityToDtoStateProvider.php
+++ b/src/State/EntityToDtoStateProvider.php
@@ -19,6 +19,6 @@ class EntityToDtoStateProvider implements ProviderInterface
     public function provide(Operation $operation, array $uriVariables = [], array $context = []): object|array|null
     {
         $entities = $this->collectionProvider->provide($operation, $uriVariables, $context);
-        dd($entities);
+        dd(iterator_to_array($entities));
     }
 }
