diff --git a/src/State/EntityToDtoStateProvider.php b/src/State/EntityToDtoStateProvider.php
index 2fcbaebe..f6c725c2 100644
--- a/src/State/EntityToDtoStateProvider.php
+++ b/src/State/EntityToDtoStateProvider.php
@@ -2,8 +2,10 @@
 
 namespace App\State;
 
+use ApiPlatform\Doctrine\Orm\Paginator;
 use ApiPlatform\Doctrine\Orm\State\CollectionProvider;
 use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\Pagination\TraversablePaginator;
 use ApiPlatform\State\ProviderInterface;
 use App\ApiResource\UserApi;
 use Symfony\Component\DependencyInjection\Attribute\Autowire;
@@ -20,13 +22,19 @@ class EntityToDtoStateProvider implements ProviderInterface
     public function provide(Operation $operation, array $uriVariables = [], array $context = []): object|array|null
     {
         $entities = $this->collectionProvider->provide($operation, $uriVariables, $context);
+        assert($entities instanceof Paginator);
 
         $dtos = [];
         foreach ($entities as $entity) {
             $dtos[] = $this->mapEntityToDto($entity);
         }
 
-        return $dtos;
+        return new TraversablePaginator(
+            new \ArrayIterator($dtos),
+            $entities->getCurrentPage(),
+            $entities->getItemsPerPage(),
+            $entities->getTotalItems()
+        );
     }
 
     private function mapEntityToDto(object $entity): object
