diff --git a/src/State/DragonTreasureSetOwnerProcessor.php b/src/State/DragonTreasureStateProcessor.php
similarity index 93%
rename from src/State/DragonTreasureSetOwnerProcessor.php
rename to src/State/DragonTreasureStateProcessor.php
index 3a745d3e..80291354 100644
--- a/src/State/DragonTreasureSetOwnerProcessor.php
+++ b/src/State/DragonTreasureStateProcessor.php
@@ -9,7 +9,7 @@ use Symfony\Bundle\SecurityBundle\Security;
 use Symfony\Component\DependencyInjection\Attribute\AsDecorator;
 
 #[AsDecorator('api_platform.doctrine.orm.state.persist_processor')]
-class DragonTreasureSetOwnerProcessor implements ProcessorInterface
+class DragonTreasureStateProcessor implements ProcessorInterface
 {
     public function __construct(private ProcessorInterface $innerProcessor, private Security $security)
     {
