diff --git a/src/State/DragonTreasureStateProcessor.php b/src/State/DragonTreasureStateProcessor.php
index 935113d5..f8a0c025 100644
--- a/src/State/DragonTreasureStateProcessor.php
+++ b/src/State/DragonTreasureStateProcessor.php
@@ -20,6 +20,7 @@ class DragonTreasureStateProcessor implements ProcessorInterface
 
     public function process(mixed $data, Operation $operation, array $uriVariables = [], array $context = [])
     {
+        dd($context);
         assert($data instanceof DragonTreasure);
         $data->setOwner($this->security->getUser());
 
