diff --git a/src/State/DragonTreasureStateProcessor.php b/src/State/DragonTreasureStateProcessor.php
index 96760f3..286684f 100644
--- a/src/State/DragonTreasureStateProcessor.php
+++ b/src/State/DragonTreasureStateProcessor.php
@@ -5,6 +5,8 @@ namespace App\State;
 use ApiPlatform\Metadata\Operation;
 use ApiPlatform\State\ProcessorInterface;
 use App\Entity\DragonTreasure;
+use App\Entity\Notification;
+use Doctrine\ORM\EntityManagerInterface;
 use Symfony\Bundle\SecurityBundle\Security;
 use Symfony\Component\DependencyInjection\Attribute\Autowire;
 
@@ -13,7 +15,8 @@ class DragonTreasureStateProcessor implements ProcessorInterface
     public function __construct(
         #[Autowire(service: 'api_platform.doctrine.orm.state.persist_processor')]
         private ProcessorInterface $innerProcessor,
-        private Security $security
+        private Security $security,
+        private EntityManagerInterface $entityManager
     )
     {
     }
@@ -32,6 +35,11 @@ class DragonTreasureStateProcessor implements ProcessorInterface
             && $data->getIsPublished()
             && $previousData->getIsPublished() !== $data->getIsPublished()
         ) {
+            $notification = new Notification();
+            $notification->setDragonTreasure($data);
+            $notification->setMessage('Treasure has been published!');
+            $this->entityManager->persist($notification);
+            $this->entityManager->flush();
         }
 
         return $data;
