diff --git a/tests/Functional/DragonTreasureResourceTest.php b/tests/Functional/DragonTreasureResourceTest.php
index 1c439b6..c3bf072 100644
--- a/tests/Functional/DragonTreasureResourceTest.php
+++ b/tests/Functional/DragonTreasureResourceTest.php
@@ -16,7 +16,12 @@ class DragonTreasureResourceTest extends ApiTestCase
 
     public function testGetCollectionOfTreasures(): void
     {
-        DragonTreasureFactory::createMany(5);
+        DragonTreasureFactory::createMany(5, [
+            'isPublished' => true,
+        ]);
+        DragonTreasureFactory::createOne([
+            'isPublished' => false,
+        ]);
 
         $json = $this->browser()
             ->get('/api/treasures')
