diff --git a/src/ApiPlatform/DragonTreasureIsPublishedExtension.php b/src/ApiPlatform/DragonTreasureIsPublishedExtension.php
new file mode 100644
index 0000000..f91e281
--- /dev/null
+++ b/src/ApiPlatform/DragonTreasureIsPublishedExtension.php
@@ -0,0 +1,16 @@
+<?php
+
+namespace App\ApiPlatform;
+
+use ApiPlatform\Doctrine\Orm\Extension\QueryCollectionExtensionInterface;
+use ApiPlatform\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use ApiPlatform\Metadata\Operation;
+use Doctrine\ORM\QueryBuilder;
+
+class DragonTreasureIsPublishedExtension implements QueryCollectionExtensionInterface
+{
+    public function applyToCollection(QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, string $resourceClass, Operation $operation = null, array $context = []): void
+    {
+        // TODO: Implement applyToCollection() method.
+    }
+}
