diff --git a/src/Entity/User.php b/src/Entity/User.php
index f66b098..afd5b35 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -49,6 +49,7 @@ class User implements UserInterface, PasswordAuthenticatedUserInterface
     private ?string $username = null;
 
     #[ORM\OneToMany(mappedBy: 'owner', targetEntity: DragonTreasure::class)]
+    #[Groups(['user:read'])]
     private Collection $dragonTreasures;
 
     public function __construct()
