diff --git a/src/State/EntityToDtoStateProvider.php b/src/State/EntityToDtoStateProvider.php
index 48cf0ff..de489de 100644
--- a/src/State/EntityToDtoStateProvider.php
+++ b/src/State/EntityToDtoStateProvider.php
@@ -34,7 +34,6 @@ class EntityToDtoStateProvider implements ProviderInterface
             foreach ($entities as $entity) {
                 $dtos[] = $this->mapEntityToDto($entity, $resourceClass);
             }
-            dd($dtos);
 
             return new TraversablePaginator(
                 new \ArrayIterator($dtos),
