diff --git a/src/Mapper/UserApiToEntityMapper.php b/src/Mapper/UserApiToEntityMapper.php
index 726d47f7..aad668d5 100644
--- a/src/Mapper/UserApiToEntityMapper.php
+++ b/src/Mapper/UserApiToEntityMapper.php
@@ -7,6 +7,7 @@ use App\Entity\DragonTreasure;
 use App\Entity\User;
 use App\Repository\UserRepository;
 use Symfony\Component\PasswordHasher\Hasher\UserPasswordHasherInterface;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
 use Symfonycasts\MicroMapper\AsMapper;
 use Symfonycasts\MicroMapper\MapperInterface;
 use Symfonycasts\MicroMapper\MicroMapperInterface;
@@ -18,6 +19,7 @@ class UserApiToEntityMapper implements MapperInterface
         private UserRepository $userRepository,
         private UserPasswordHasherInterface $userPasswordHasher,
         private MicroMapperInterface $microMapper,
+        private PropertyAccessorInterface $propertyAccessor,
     )
     {
     }
@@ -54,7 +56,8 @@ class UserApiToEntityMapper implements MapperInterface
                 MicroMapperInterface::MAX_DEPTH => 0,
             ]);
         }
-        dd($dragonTreasureEntities);
+        $this->propertyAccessor->setValue($entity, 'dragonTreasures', $dragonTreasureEntities);
+        dd($entity);
 
         return $entity;
     }
