diff --git a/src/Validator/TreasuresAllowedOwnerChangeValidator.php b/src/Validator/TreasuresAllowedOwnerChangeValidator.php
index 868324ad..ed30707f 100644
--- a/src/Validator/TreasuresAllowedOwnerChangeValidator.php
+++ b/src/Validator/TreasuresAllowedOwnerChangeValidator.php
@@ -2,17 +2,13 @@
 
 namespace App\Validator;
 
-use App\Entity\DragonTreasure;
-use Doctrine\ORM\EntityManagerInterface;
+use App\ApiResource\DragonTreasureApi;
+use App\ApiResource\UserApi;
 use Symfony\Component\Validator\Constraint;
 use Symfony\Component\Validator\ConstraintValidator;
 
 class TreasuresAllowedOwnerChangeValidator extends ConstraintValidator
 {
-    public function __construct(private EntityManagerInterface $entityManager)
-    {
-    }
-
     public function validate($value, Constraint $constraint): void
     {
         assert($constraint instanceof TreasuresAllowedOwnerChange);
@@ -21,15 +17,13 @@ class TreasuresAllowedOwnerChangeValidator extends ConstraintValidator
             return;
         }
 
-        dd($value);
+        assert($value instanceof UserApi);
 
-        $unitOfWork = $this->entityManager->getUnitOfWork();
-        foreach ($value as $dragonTreasure) {
-            assert($dragonTreasure instanceof DragonTreasure);
+        foreach ($value->dragonTreasures as $dragonTreasureApi) {
+            assert($dragonTreasureApi instanceof DragonTreasureApi);
 
-            $originalData = $unitOfWork->getOriginalEntityData($dragonTreasure);
-            $originalOwnerId = $originalData['owner_id'];
-            $newOwnerId = $dragonTreasure->getOwner()->getId();
+            $originalOwnerId = $dragonTreasureApi->owner?->id;
+            $newOwnerId = $value->id;
 
             if (!$originalOwnerId || $originalOwnerId === $newOwnerId) {
                 return;
