diff --git a/tests/Functional/UserResourceTest.php b/tests/Functional/UserResourceTest.php
index 1c42b10a..523680f1 100644
--- a/tests/Functional/UserResourceTest.php
+++ b/tests/Functional/UserResourceTest.php
@@ -83,7 +83,6 @@ class UserResourceTest extends ApiTestCase
         $user = UserFactory::createOne();
         $otherUser = UserFactory::createOne();
         $dragonTreasure = DragonTreasureFactory::createOne(['owner' => $otherUser]);
-        dump('Real owner is ' . $otherUser->getId());
 
         $this->browser()
             ->actingAs($user)
