diff --git a/src/Mapper/UserApiToEntityMapper.php b/src/Mapper/UserApiToEntityMapper.php
index ca11a83..bf1474a 100644
--- a/src/Mapper/UserApiToEntityMapper.php
+++ b/src/Mapper/UserApiToEntityMapper.php
@@ -44,6 +44,7 @@ class UserApiToEntityMapper implements MapperInterface
         if ($dto->password) {
             $entity->setPassword($this->userPasswordHasher->hashPassword($entity, $dto->password));
         }
+        dump($dto);
         // TODO dragonTreasures if we change them to writeable
 
         return $entity;
