diff --git a/src/Validator/TreasuresAllowedOwnerChange.php b/src/Validator/TreasuresAllowedOwnerChange.php
index bd65e4c..a901252 100644
--- a/src/Validator/TreasuresAllowedOwnerChange.php
+++ b/src/Validator/TreasuresAllowedOwnerChange.php
@@ -4,11 +4,6 @@ namespace App\Validator;
 
 use Symfony\Component\Validator\Constraint;
 
-/**
- * @Annotation
- *
- * @Target({"PROPERTY", "METHOD", "ANNOTATION"})
- */
 #[\Attribute(\Attribute::TARGET_CLASS | \Attribute::IS_REPEATABLE)]
 class TreasuresAllowedOwnerChange extends Constraint
 {
