diff --git a/src/Mapper/UserEntityToApiMapper.php b/src/Mapper/UserEntityToApiMapper.php
index 15fbd857..bdc2f0b7 100644
--- a/src/Mapper/UserEntityToApiMapper.php
+++ b/src/Mapper/UserEntityToApiMapper.php
@@ -41,7 +41,7 @@ class UserEntityToApiMapper implements MapperInterface
         $dto->username = $entity->getUsername();
         $dto->dragonTreasures = array_map(function(DragonTreasure $dragonTreasure) {
             return $this->microMapper->map($dragonTreasure, DragonTreasureApi::class, [
-                MicroMapperInterface::MAX_DEPTH => 1,
+                MicroMapperInterface::MAX_DEPTH => 0,
             ]);
         }, $entity->getPublishedDragonTreasures()->getValues());
         $dto->flameThrowingDistance = rand(1, 100);
diff --git a/src/State/EntityClassDtoStateProvider.php b/src/State/EntityClassDtoStateProvider.php
index a1b4779c..f72e9003 100644
--- a/src/State/EntityClassDtoStateProvider.php
+++ b/src/State/EntityClassDtoStateProvider.php
@@ -34,6 +34,7 @@ class EntityClassDtoStateProvider implements ProviderInterface
             foreach ($entities as $entity) {
                 $dtos[] = $this->mapEntityToDto($entity, $resourceClass);
             }
+            dd($dtos);
 
             return new TraversablePaginator(
                 new \ArrayIterator($dtos),
