diff --git a/src/Mapper/DragonTreasureEntityToApiMapper.php b/src/Mapper/DragonTreasureEntityToApiMapper.php
index b307c10..52b98c8 100644
--- a/src/Mapper/DragonTreasureEntityToApiMapper.php
+++ b/src/Mapper/DragonTreasureEntityToApiMapper.php
@@ -39,7 +39,9 @@ class DragonTreasureEntityToApiMapper implements MapperInterface
         assert($dto instanceof DragonTreasureApi);
 
         $dto->name = $entity->getName();
-        $dto->owner = $this->microMapper->map($entity->getOwner(), UserApi::class);
+        $dto->owner = $this->microMapper->map($entity->getOwner(), UserApi::class, [
+            MicroMapperInterface::MAX_DEPTH => 0,
+        ]);
         $dto->description = $entity->getDescription();
         $dto->value = $entity->getValue();
         $dto->coolFactor = $entity->getCoolFactor();
