diff --git a/src/Validator/TreasuresAllowedOwnerChange.php b/src/Validator/TreasuresAllowedOwnerChange.php
index 2ae61c48..bd65e4c6 100644
--- a/src/Validator/TreasuresAllowedOwnerChange.php
+++ b/src/Validator/TreasuresAllowedOwnerChange.php
@@ -9,7 +9,7 @@ use Symfony\Component\Validator\Constraint;
  *
  * @Target({"PROPERTY", "METHOD", "ANNOTATION"})
  */
-#[\Attribute(\Attribute::TARGET_PROPERTY | \Attribute::TARGET_METHOD | \Attribute::IS_REPEATABLE)]
+#[\Attribute(\Attribute::TARGET_CLASS | \Attribute::IS_REPEATABLE)]
 class TreasuresAllowedOwnerChange extends Constraint
 {
     /*
@@ -17,4 +17,9 @@ class TreasuresAllowedOwnerChange extends Constraint
      * Then, use these in your validator class.
      */
     public string $message = 'One of the treasures illegally changed owners.';
+
+    public function getTargets(): string|array
+    {
+        return self::CLASS_CONSTRAINT;
+    }
 }
