diff --git a/src/ApiResource/UserApi.php b/src/ApiResource/UserApi.php
index d74b9a05..85f4a1eb 100644
--- a/src/ApiResource/UserApi.php
+++ b/src/ApiResource/UserApi.php
@@ -12,7 +12,6 @@ use ApiPlatform\Metadata\Get;
 use ApiPlatform\Metadata\GetCollection;
 use ApiPlatform\Metadata\Patch;
 use ApiPlatform\Metadata\Post;
-use App\Entity\DragonTreasure;
 use App\State\EntityClassDtoStateProcessor;
 use App\State\EntityToDtoStateProvider;
 use App\Entity\User;
@@ -61,7 +60,7 @@ class UserApi
     public ?string $password = null;
 
     /**
-     * @var array<int, DragonTreasure>
+     * @var array<int, DragonTreasureApi>
      */
     #[ApiProperty(writable: false)]
     public array $dragonTreasures = [];
