diff --git a/tests/Functional/DragonTreasureResourceTest.php b/tests/Functional/DragonTreasureResourceTest.php
index 33a5d3a0..3d266fe7 100644
--- a/tests/Functional/DragonTreasureResourceTest.php
+++ b/tests/Functional/DragonTreasureResourceTest.php
@@ -74,6 +74,7 @@ class DragonTreasureResourceTest extends ApiTestCase
                 'description' => 'It sparkles when I wave it in the air.',
                 'value' => 1000,
                 'coolFactor' => 5,
+                'owner' => '/api/users/'.$user->getId(),
             ]))
             ->assertStatus(201)
             ->assertJsonMatches('name', 'A shiny thing')
