diff --git a/src/Mapper/UserEntityToApiMapper.php b/src/Mapper/UserEntityToApiMapper.php
index 3cf5199..15fbd85 100644
--- a/src/Mapper/UserEntityToApiMapper.php
+++ b/src/Mapper/UserEntityToApiMapper.php
@@ -40,7 +40,9 @@ class UserEntityToApiMapper implements MapperInterface
         $dto->email = $entity->getEmail();
         $dto->username = $entity->getUsername();
         $dto->dragonTreasures = array_map(function(DragonTreasure $dragonTreasure) {
-            return $this->microMapper->map($dragonTreasure, DragonTreasureApi::class);
+            return $this->microMapper->map($dragonTreasure, DragonTreasureApi::class, [
+                MicroMapperInterface::MAX_DEPTH => 1,
+            ]);
         }, $entity->getPublishedDragonTreasures()->getValues());
         $dto->flameThrowingDistance = rand(1, 100);
 
