diff --git a/src/ApiResource/UserApi.php b/src/ApiResource/UserApi.php
index 7ed8c2ab..11af438d 100644
--- a/src/ApiResource/UserApi.php
+++ b/src/ApiResource/UserApi.php
@@ -41,6 +41,7 @@ use Symfony\Component\Validator\Constraints as Assert;
 #[ApiFilter(SearchFilter::class, properties: [
     'username' => 'partial',
 ])]
+#[TreasuresAllowedOwnerChange]
 class UserApi
 {
     #[ApiProperty(readable: false, writable: false, identifier: true)]
@@ -63,7 +64,6 @@ class UserApi
     /**
      * @var array<int, DragonTreasureApi>
      */
-    #[TreasuresAllowedOwnerChange]
     public array $dragonTreasures = [];
 
     #[ApiProperty(writable: false)]
diff --git a/src/Validator/TreasuresAllowedOwnerChangeValidator.php b/src/Validator/TreasuresAllowedOwnerChangeValidator.php
index f712ef17..868324ad 100644
--- a/src/Validator/TreasuresAllowedOwnerChangeValidator.php
+++ b/src/Validator/TreasuresAllowedOwnerChangeValidator.php
@@ -21,7 +21,7 @@ class TreasuresAllowedOwnerChangeValidator extends ConstraintValidator
             return;
         }
 
-        return;
+        dd($value);
 
         $unitOfWork = $this->entityManager->getUnitOfWork();
         foreach ($value as $dragonTreasure) {
