diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index 6477d85..ee6c76f 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -34,7 +34,9 @@ use function Symfony\Component\String\u;
             ],
         ),
         new GetCollection(),
-        new Post(),
+        new Post(
+            security: 'is_granted("ROLE_TREASURE_CREATE")',
+        ),
         new Put(),
         new Patch(),
     ],
