diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index 469af30..65d4d96 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -65,8 +65,9 @@ class DragonTreasure
     #[ORM\Column]
     private bool $isPublished = false;
 
-    public function __construct()
+    public function __construct(string $name)
     {
+        $this->name = $name;
         $this->plunderedAt = new \DateTimeImmutable();
     }
 
@@ -80,13 +81,6 @@ class DragonTreasure
         return $this->name;
     }
 
-    public function setName(string $name): self
-    {
-        $this->name = $name;
-
-        return $this;
-    }
-
     public function getDescription(): ?string
     {
         return $this->description;
