diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index 65d4d96..0c64f71 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -65,7 +65,7 @@ class DragonTreasure
     #[ORM\Column]
     private bool $isPublished = false;
 
-    public function __construct(string $name)
+    public function __construct(string $name = null)
     {
         $this->name = $name;
         $this->plunderedAt = new \DateTimeImmutable();
