diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index a5f7d0d..65d4d96 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -65,9 +65,9 @@ class DragonTreasure
     #[ORM\Column]
     private bool $isPublished = false;
 
-    public function __construct(string $treasureName)
+    public function __construct(string $name)
     {
-        $this->name = $treasureName;
+        $this->name = $name;
         $this->plunderedAt = new \DateTimeImmutable();
     }
 
