diff --git a/src/ApiResource/UserApi.php b/src/ApiResource/UserApi.php
index 9a3f21e..f9984ca 100644
--- a/src/ApiResource/UserApi.php
+++ b/src/ApiResource/UserApi.php
@@ -4,7 +4,9 @@ namespace App\ApiResource;
 
 use ApiPlatform\Metadata\ApiResource;
 
-#[ApiResource]
+#[ApiResource(
+    provider: 'api_platform.doctrine.orm.state.collection_provider',
+)]
 class UserApi
 {
     public ?int $id = null;
