diff --git a/src/ApiResource/UserApi.php b/src/ApiResource/UserApi.php
index b27b1f0a..ba4bb75b 100644
--- a/src/ApiResource/UserApi.php
+++ b/src/ApiResource/UserApi.php
@@ -2,10 +2,12 @@
 
 namespace App\ApiResource;
 
+use ApiPlatform\Doctrine\Orm\State\CollectionProvider;
 use ApiPlatform\Metadata\ApiResource;
 
 #[ApiResource(
     shortName: 'User',
+    provider: CollectionProvider::class,
 )]
 class UserApi
 {
