diff --git a/tests/Functional/UserResourceTest.php b/tests/Functional/UserResourceTest.php
index 0e5a4e1..cf35bdf 100644
--- a/tests/Functional/UserResourceTest.php
+++ b/tests/Functional/UserResourceTest.php
@@ -40,6 +40,7 @@ class UserResourceTest extends ApiTestCase
                'json' => [
                    'username' => 'changed',
                ],
+               'headers' => ['Content-Type' => 'application/merge-patch+json']
            ])
            ->assertStatus(200)
        ;
