diff --git a/src/ApiResource/UserApi.php b/src/ApiResource/UserApi.php
index 59c4f2b..9216915 100644
--- a/src/ApiResource/UserApi.php
+++ b/src/ApiResource/UserApi.php
@@ -5,6 +5,8 @@ namespace App\ApiResource;
 use ApiPlatform\Doctrine\Orm\State\Options;
 use ApiPlatform\Metadata\ApiResource;
 use App\Entity\User;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
 
 #[ApiResource(
     stateOptions: new Options(entityClass: User::class),
@@ -16,4 +18,11 @@ class UserApi
     public ?string $email = null;
 
     public ?string $username = null;
+
+    public Collection $dragonTreasures;
+
+    public function __construct()
+    {
+        $this->dragonTreasures = new ArrayCollection();
+    }
 }
diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index c59bbc3..070040e 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -145,7 +145,7 @@ class DragonTreasure
     /**
      * @var bool Non-persisted property to help determine if the treasure is owned by the authenticated user
      */
-    private bool $isOwnedByAuthenticatedUser;
+    private bool $isOwnedByAuthenticatedUser = false;
 
     public function __construct(string $name = null)
     {
