diff --git a/src/Entity/User.php b/src/Entity/User.php
index 905175e..83b81f6 100644
--- a/src/Entity/User.php
+++ b/src/Entity/User.php
@@ -88,6 +88,7 @@ class User implements UserInterface, PasswordAuthenticatedUserInterface
 
     #[Groups(['user:write'])]
     #[SerializedName('password')]
+    #[Assert\NotBlank]
     private ?string $plainPassword = null;
 
     #[ORM\Column(length: 255, unique: true)]
