diff --git a/src/ApiResource/UserApi.php b/src/ApiResource/UserApi.php
index 482f5e3..c300d1b 100644
--- a/src/ApiResource/UserApi.php
+++ b/src/ApiResource/UserApi.php
@@ -2,7 +2,9 @@
 
 namespace App\ApiResource;
 
+use ApiPlatform\Doctrine\Orm\Filter\SearchFilter;
 use ApiPlatform\Doctrine\Orm\State\Options;
+use ApiPlatform\Metadata\ApiFilter;
 use ApiPlatform\Metadata\ApiResource;
 use App\Entity\User;
 
@@ -11,6 +13,9 @@ use App\Entity\User;
     provider: 'api_platform.doctrine.orm.state.collection_provider',
     stateOptions: new Options(entityClass: User::class),
 )]
+#[ApiFilter(SearchFilter::class, properties: [
+    'username' => 'partial',
+])]
 class UserApi
 {
     public int $id;
