diff --git a/src/ApiResource/UserApi.php b/src/ApiResource/UserApi.php
index 5ea5287..e502f61 100644
--- a/src/ApiResource/UserApi.php
+++ b/src/ApiResource/UserApi.php
@@ -7,7 +7,6 @@ use ApiPlatform\Metadata\ApiResource;
 use App\Entity\User;
 
 #[ApiResource(
-    provider: 'api_platform.doctrine.orm.state.collection_provider',
     stateOptions: new Options(entityClass: User::class),
 )]
 class UserApi
