diff --git a/src/ApiResource/UserApi.php b/src/ApiResource/UserApi.php
index 1aff161..a09ab2c 100644
--- a/src/ApiResource/UserApi.php
+++ b/src/ApiResource/UserApi.php
@@ -2,7 +2,9 @@
 
 namespace App\ApiResource;
 
+use ApiPlatform\Doctrine\Orm\Filter\SearchFilter;
 use ApiPlatform\Doctrine\Orm\State\Options;
+use ApiPlatform\Metadata\ApiFilter;
 use ApiPlatform\Metadata\ApiResource;
 use App\Entity\DragonTreasure;
 use App\Entity\User;
@@ -13,6 +15,9 @@ use Doctrine\Common\Collections\Collection;
     paginationItemsPerPage: 5,
     stateOptions: new Options(entityClass: User::class),
 )]
+#[ApiFilter(SearchFilter::class, properties: [
+    'username' => 'partial',
+])]
 class UserApi
 {
     public ?int $id = null;
