diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index 3ac7f54..afe217c 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -52,6 +52,10 @@ use function Symfony\Component\String\u;
     ],
     paginationItemsPerPage: 10,
 )]
+#[ApiResource(
+    uriTemplate: '/users/{user_id}/treasures.{_format}',
+    operations: [new GetCollection()],
+)]
 #[ApiFilter(PropertyFilter::class)]
 #[ApiFilter(SearchFilter::class, properties: [
     'owner.username' => 'partial',
