diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index afe217c..3e8b1e1 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -54,6 +54,7 @@ use function Symfony\Component\String\u;
 )]
 #[ApiResource(
     uriTemplate: '/users/{user_id}/treasures.{_format}',
+    shortName: 'Treasure',
     operations: [new GetCollection()],
 )]
 #[ApiFilter(PropertyFilter::class)]
