diff --git a/tests/Functional/DragonTreasureResourceTest.php b/tests/Functional/DragonTreasureResourceTest.php
index 751e374..e8ff1d2 100644
--- a/tests/Functional/DragonTreasureResourceTest.php
+++ b/tests/Functional/DragonTreasureResourceTest.php
@@ -2,6 +2,7 @@
 
 namespace App\Tests\Functional;
 
+use App\Factory\DragonTreasureFactory;
 use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
 use Zenstruck\Browser\Test\HasBrowser;
 use Zenstruck\Foundry\Test\ResetDatabase;
@@ -13,11 +14,14 @@ class DragonTreasureResourceTest extends KernelTestCase
 
     public function testGetCollectionOfTreasures(): void
     {
+        DragonTreasureFactory::createMany(5);
+
         $this->browser()
             ->get('/api/treasures')
             ->dump()
             ->assertJson()
-            ->assertJsonMatches('"hydra:totalItems"', 0)
+            ->assertJsonMatches('"hydra:totalItems"', 5)
+            ->assertJsonMatches('length("hydra:member")', 5)
         ;
     }
 }
