diff --git a/tests/Functional/DragonTreasureResourceTest.php b/tests/Functional/DragonTreasureResourceTest.php
index b679d30..751e374 100644
--- a/tests/Functional/DragonTreasureResourceTest.php
+++ b/tests/Functional/DragonTreasureResourceTest.php
@@ -17,7 +17,7 @@ class DragonTreasureResourceTest extends KernelTestCase
             ->get('/api/treasures')
             ->dump()
             ->assertJson()
-            ->assertJsonMatches('hydra:totalItems', 0)
+            ->assertJsonMatches('"hydra:totalItems"', 0)
         ;
     }
 }
