diff --git a/tests/Functional/DragonTreasureResourceTest.php b/tests/Functional/DragonTreasureResourceTest.php
index 7cc1bd3..ae9bd55 100644
--- a/tests/Functional/DragonTreasureResourceTest.php
+++ b/tests/Functional/DragonTreasureResourceTest.php
@@ -16,6 +16,7 @@ class DragonTreasureResourceTest extends KernelTestCase
         $this->browser()
             ->get('/api/treasures')
             ->dump()
+            ->assertJson()
         ;
     }
 }
