diff --git a/tests/Functional/DragonTreasureResourceTest.php b/tests/Functional/DragonTreasureResourceTest.php
index d9926cc..f820691 100644
--- a/tests/Functional/DragonTreasureResourceTest.php
+++ b/tests/Functional/DragonTreasureResourceTest.php
@@ -25,7 +25,7 @@ class DragonTreasureResourceTest extends KernelTestCase
             ->json()
         ;
 
-        $json->assertMatches('keys("hydra:member"[0])', [
+        $this->assertSame(array_keys($json->decoded()['hydra:member'][0]), [
             '@id',
             '@type',
             'name',
