diff --git a/src/ApiResource/DragonTreasureApi.php b/src/ApiResource/DragonTreasureApi.php
index 8edd1a1d..c946e221 100644
--- a/src/ApiResource/DragonTreasureApi.php
+++ b/src/ApiResource/DragonTreasureApi.php
@@ -56,6 +56,8 @@ class DragonTreasureApi
     #[LessThanOrEqual(10)]
     public ?int $coolFactor = 0;
 
+    public bool $isPublished = false;
+
     #[IsValidOwner]
     public ?UserApi $owner = null;
 
diff --git a/src/Mapper/DragonTreasureApiToEntityMapper.php b/src/Mapper/DragonTreasureApiToEntityMapper.php
index 8f1fa73d..1db88d7c 100644
--- a/src/Mapper/DragonTreasureApiToEntityMapper.php
+++ b/src/Mapper/DragonTreasureApiToEntityMapper.php
@@ -49,8 +49,7 @@ class DragonTreasureApiToEntityMapper implements MapperInterface
         $entity->setDescription($dto->description);
         $entity->setCoolFactor($dto->coolFactor);
         $entity->setValue($dto->value);
-
-        // TODO: set published
+        $entity->setIsPublished($dto->isPublished);
 
         return $entity;
     }
diff --git a/src/Mapper/DragonTreasureEntityToApiMapper.php b/src/Mapper/DragonTreasureEntityToApiMapper.php
index 52b98c88..0c491d9b 100644
--- a/src/Mapper/DragonTreasureEntityToApiMapper.php
+++ b/src/Mapper/DragonTreasureEntityToApiMapper.php
@@ -39,6 +39,7 @@ class DragonTreasureEntityToApiMapper implements MapperInterface
         assert($dto instanceof DragonTreasureApi);
 
         $dto->name = $entity->getName();
+        $dto->isPublished = $entity->getIsPublished();
         $dto->owner = $this->microMapper->map($entity->getOwner(), UserApi::class, [
             MicroMapperInterface::MAX_DEPTH => 0,
         ]);
