diff --git a/src/Automapper/EntityToUserApiMapperConfiguration.php b/src/Automapper/EntityToUserApiMapperConfiguration.php
index 8067e18..8d2e031 100644
--- a/src/Automapper/EntityToUserApiMapperConfiguration.php
+++ b/src/Automapper/EntityToUserApiMapperConfiguration.php
@@ -15,7 +15,9 @@ class EntityToUserApiMapperConfiguration implements MapperConfigurationInterface
     {
         assert($metadata instanceof MapperMetadata);
 
-        $metadata->forMember('dragonTreasures', fn (User $user) => $user->getPublishedDragonTreasures()->getValues());
+        $metadata->forMember('dragonTreasures', function (User $user) {
+            return $user->getPublishedDragonTreasures()->getValues();
+        });
         $metadata->forMember('flameThrowingDistance', fn (User $user) => rand(0, 10));
         // always start the plain password with null
         $metadata->forMember('password', fn (User $user) => null);
