diff --git a/tests/Functional/DragonTreasureResourceTest.php b/tests/Functional/DragonTreasureResourceTest.php
index abbb970..33a5d3a 100644
--- a/tests/Functional/DragonTreasureResourceTest.php
+++ b/tests/Functional/DragonTreasureResourceTest.php
@@ -125,6 +125,7 @@ class DragonTreasureResourceTest extends ApiTestCase
                 'json' => [
                     'value' => 12345,
                 ],
+                'headers' => ['Content-Type' => 'application/merge-patch+json']
             ])
             ->assertStatus(200)
             ->assertJsonMatches('value', 12345)
@@ -139,6 +140,7 @@ class DragonTreasureResourceTest extends ApiTestCase
                     // be tricky and try to change the owner
                     'owner' => '/api/users/'.$user2->getId(),
                 ],
+                'headers' => ['Content-Type' => 'application/merge-patch+json']
             ])
             ->assertStatus(403)
         ;
@@ -150,6 +152,7 @@ class DragonTreasureResourceTest extends ApiTestCase
                     // change the owner to someone else
                     'owner' => '/api/users/'.$user2->getId(),
                 ],
+                'headers' => ['Content-Type' => 'application/merge-patch+json']
             ])
             ->assertStatus(422)
         ;
@@ -169,6 +172,7 @@ class DragonTreasureResourceTest extends ApiTestCase
                 'json' => [
                     'value' => 12345,
                 ],
+                'headers' => ['Content-Type' => 'application/merge-patch+json']
             ])
             ->assertStatus(200)
             ->assertJsonMatches('value', 12345)
@@ -189,6 +193,7 @@ class DragonTreasureResourceTest extends ApiTestCase
                 'json' => [
                     'value' => 12345,
                 ],
+                'headers' => ['Content-Type' => 'application/merge-patch+json']
             ])
             ->assertStatus(200)
             ->assertJsonMatches('value', 12345)
@@ -210,6 +215,7 @@ class DragonTreasureResourceTest extends ApiTestCase
                 'json' => [
                     'value' => 12345,
                 ],
+                'headers' => ['Content-Type' => 'application/merge-patch+json']
             ])
             ->assertStatus(200)
             ->assertJsonMatches('value', 12345)
@@ -232,6 +238,7 @@ class DragonTreasureResourceTest extends ApiTestCase
                 'json' => [
                     'isPublished' => true,
                 ],
+                'headers' => ['Content-Type' => 'application/merge-patch+json']
             ])
             ->assertStatus(200)
             ->assertJsonMatches('isPublished', true)
