diff --git a/src/Security/Voter/DragonTreasureVoter.php b/src/Security/Voter/DragonTreasureApiVoter.php
similarity index 97%
rename from src/Security/Voter/DragonTreasureVoter.php
rename to src/Security/Voter/DragonTreasureApiVoter.php
index 9c5254d..98a8b6a 100644
--- a/src/Security/Voter/DragonTreasureVoter.php
+++ b/src/Security/Voter/DragonTreasureApiVoter.php
@@ -8,7 +8,7 @@ use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
 use Symfony\Component\Security\Core\Authorization\Voter\Voter;
 use Symfony\Component\Security\Core\User\UserInterface;
 
-class DragonTreasureVoter extends Voter
+class DragonTreasureApiVoter extends Voter
 {
     public const EDIT = 'EDIT';
 
