diff --git a/src/ApiResource/DragonTreasureApi.php b/src/ApiResource/DragonTreasureApi.php
index a89aaba..25dc27d 100644
--- a/src/ApiResource/DragonTreasureApi.php
+++ b/src/ApiResource/DragonTreasureApi.php
@@ -6,6 +6,7 @@ use ApiPlatform\Doctrine\Orm\State\Options;
 use ApiPlatform\Metadata\ApiProperty;
 use ApiPlatform\Metadata\ApiResource;
 use App\Entity\DragonTreasure;
+use Symfony\Component\Validator\Constraints\NotBlank;
 use App\State\EntityClassDtoStateProcessor;
 use App\State\EntityClassDtoStateProvider;
 
@@ -21,6 +22,7 @@ class DragonTreasureApi
     #[ApiProperty(readable: false, writable: false, identifier: true)]
     public int $id;
 
+    #[NotBlank]
     public ?string $name = null;
 
     public ?UserApi $owner = null;
diff --git a/tests/Functional/DragonTreasureResourceTest.php b/tests/Functional/DragonTreasureResourceTest.php
index d61e836..21f67eb 100644
--- a/tests/Functional/DragonTreasureResourceTest.php
+++ b/tests/Functional/DragonTreasureResourceTest.php
@@ -59,7 +59,7 @@ class DragonTreasureResourceTest extends ApiTestCase
             ->assertStatus(404);
     }
 
-    public function testPostToCreateTreasure(): void
+    public function testPostToCreateTreasureWithLogin(): void
     {
         $user = UserFactory::createOne();
 
@@ -74,6 +74,7 @@ class DragonTreasureResourceTest extends ApiTestCase
                 'description' => 'It sparkles when I wave it in the air.',
                 'value' => 1000,
                 'coolFactor' => 5,
+                'owner' => '/api/users/'.$user->getId(),
             ]))
             ->assertStatus(201)
             ->assertJsonMatches('name', 'A shiny thing')
