diff --git a/src/Automapper/DragonTreasureApiToEntityMapper.php b/src/Automapper/DragonTreasureApiToEntityMapper.php
index 46cb8c62..0890a732 100644
--- a/src/Automapper/DragonTreasureApiToEntityMapper.php
+++ b/src/Automapper/DragonTreasureApiToEntityMapper.php
@@ -7,12 +7,14 @@ use App\Entity\DragonTreasure;
 use App\MicroMapper\AsMapper;
 use App\MicroMapper\MapperInterface;
 use App\Repository\DragonTreasureRepository;
+use Symfony\Bundle\SecurityBundle\Security;
 
 #[AsMapper(from: DragonTreasureApi::class, to: DragonTreasure::class)]
 class DragonTreasureApiToEntityMapper implements MapperInterface
 {
     public function __construct(
         private DragonTreasureRepository $repository,
+        private Security $security,
     )
     {
 
@@ -38,6 +40,12 @@ class DragonTreasureApiToEntityMapper implements MapperInterface
         assert($dto instanceof DragonTreasureApi);
         assert($entity instanceof DragonTreasure);
 
+        if ($dto->owner) {
+            dd($dto->owner);
+        } else {
+            $entity->setOwner($this->security->getUser());
+        }
+
         // TODO owner if needed
         // TODO and other fields
 
