diff --git a/src/State/DragonTreasureStateProcessor.php b/src/State/DragonTreasureStateProcessor.php
index ec6ef00..fad80d3 100644
--- a/src/State/DragonTreasureStateProcessor.php
+++ b/src/State/DragonTreasureStateProcessor.php
@@ -15,6 +15,8 @@ class DragonTreasureStateProcessor implements ProcessorInterface
 
     public function process(mixed $data, Operation $operation, array $uriVariables = [], array $context = [])
     {
+        dd($context['previous_data']);
+
         return $this->innerProcessor->process($data, $operation, $uriVariables, $context);
     }
 }
