diff --git a/src/Security/Voter/DragonTreasureApiVoter.php b/src/Security/Voter/DragonTreasureApiVoter.php
index 6db66e0..74b16b7 100644
--- a/src/Security/Voter/DragonTreasureApiVoter.php
+++ b/src/Security/Voter/DragonTreasureApiVoter.php
@@ -3,6 +3,7 @@
 namespace App\Security\Voter;
 
 use App\ApiResource\DragonTreasureApi;
+use App\Entity\User;
 use Symfony\Bundle\SecurityBundle\Security;
 use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
 use Symfony\Component\Security\Core\Authorization\Voter\Voter;
@@ -26,7 +27,7 @@ class DragonTreasureApiVoter extends Voter
     {
         $user = $token->getUser();
         // if the user is anonymous, do not grant access
-        if (!$user instanceof UserInterface) {
+        if (!$user instanceof User) {
             return false;
         }
 
@@ -35,7 +36,6 @@ class DragonTreasureApiVoter extends Voter
         }
 
         assert($subject instanceof DragonTreasureApi);
-        dd($subject);
 
         // ... (check conditions and return true to grant permission) ...
         switch ($attribute) {
@@ -44,7 +44,7 @@ class DragonTreasureApiVoter extends Voter
                     return false;
                 }
 
-                if ($subject->owner === $user) {
+                if ($subject->owner?->id === $user->getId()) {
                     return true;
                 }
 
