diff --git a/src/Controller/MainController.php b/src/Controller/MainController.php
index 87960a0..a6ecb9c 100644
--- a/src/Controller/MainController.php
+++ b/src/Controller/MainController.php
@@ -2,15 +2,22 @@
 
 namespace App\Controller;
 
+use App\Entity\User;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Http\Attribute\CurrentUser;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
 
 class MainController extends AbstractController
 {
     #[Route('/')]
-    public function homepage(): Response
+    public function homepage(NormalizerInterface $normalizer, #[CurrentUser] User $user = null): Response
     {
-        return $this->render('main/homepage.html.twig');
+        return $this->render('main/homepage.html.twig', [
+            'userData' => $normalizer->normalize($user, 'jsonld', [
+                'groups' => ['user:read'],
+            ]),
+        ]);
     }
 }
diff --git a/templates/main/homepage.html.twig b/templates/main/homepage.html.twig
index 933c1b6..3a6648e 100644
--- a/templates/main/homepage.html.twig
+++ b/templates/main/homepage.html.twig
@@ -3,6 +3,6 @@
 {% block body %}
     <div {{ vue_component('TreasureConnectApp', {
         entrypoint: path('api_entrypoint'),
-        user: app.user
+        user: userData,
     }) }}></div>
 {% endblock %}
