diff --git a/assets/vue/controllers/TreasureConnectApp.vue b/assets/vue/controllers/TreasureConnectApp.vue
index f082a9c..78ab9de 100644
--- a/assets/vue/controllers/TreasureConnectApp.vue
+++ b/assets/vue/controllers/TreasureConnectApp.vue
@@ -31,7 +31,7 @@ import coinLogoPath from '../../images/coinLogo.png';
 import goldPilePath from '../../images/GoldPile.png';
 
 defineProps(['entrypoint']);
-const user = ref(null);
+const user = ref(window.user);
 
 const onUserAuthenticated = async (userUri) => {
     const response = await fetch(userUri);
