diff --git a/assets/vue/controllers/TreasureConnectApp.vue b/assets/vue/controllers/TreasureConnectApp.vue
index b537331..12b04d2 100644
--- a/assets/vue/controllers/TreasureConnectApp.vue
+++ b/assets/vue/controllers/TreasureConnectApp.vue
@@ -32,7 +32,6 @@ import coinLogoPath from '../../images/coinLogo.png';
 import goldPilePath from '../../images/GoldPile.png';
 
 const props = defineProps(['entrypoint', 'user'])
-console.log(props.user);
 const user = ref(window.user);
 
 const onUserAuthenticated = async (userUri) => {
