diff --git a/src/DataFixtures/AppFixtures.php b/src/DataFixtures/AppFixtures.php
index 7137f2b..659a526 100644
--- a/src/DataFixtures/AppFixtures.php
+++ b/src/DataFixtures/AppFixtures.php
@@ -3,6 +3,7 @@
 namespace App\DataFixtures;
 
 use App\Factory\DragonTreasureFactory;
+use App\Factory\UserFactory;
 use Doctrine\Bundle\FixturesBundle\Fixture;
 use Doctrine\Persistence\ObjectManager;
 
@@ -11,5 +12,6 @@ class AppFixtures extends Fixture
     public function load(ObjectManager $manager): void
     {
         DragonTreasureFactory::createMany(40);
+        UserFactory::createMany(10);
     }
 }
