diff --git a/src/Validator/IsValidOwnerValidator.php b/src/Validator/IsValidOwnerValidator.php
index 241cc4a..f08b123 100644
--- a/src/Validator/IsValidOwnerValidator.php
+++ b/src/Validator/IsValidOwnerValidator.php
@@ -2,6 +2,7 @@
 
 namespace App\Validator;
 
+use App\Entity\User;
 use Symfony\Component\Validator\Constraint;
 use Symfony\Component\Validator\ConstraintValidator;
 
@@ -9,15 +10,16 @@ class IsValidOwnerValidator extends ConstraintValidator
 {
     public function validate($value, Constraint $constraint)
     {
-        /* @var App\Validator\IsValidOwner $constraint */
+        assert($constraint instanceof IsValidOwner);
 
         if (null === $value || '' === $value) {
             return;
         }
 
-        // TODO: implement the validation here
+        // constraint is only meant to be used above a User property
+        assert($value instanceof User);
+
         $this->context->buildViolation($constraint->message)
-            ->setParameter('{{ value }}', $value)
             ->addViolation();
     }
 }
