diff --git a/src/Validator/IsValidOwnerValidator.php b/src/Validator/IsValidOwnerValidator.php
index f8949c5..8b409c0 100644
--- a/src/Validator/IsValidOwnerValidator.php
+++ b/src/Validator/IsValidOwnerValidator.php
@@ -29,7 +29,9 @@ class IsValidOwnerValidator extends ConstraintValidator
             throw new \LogicException('IsOwnerValidator should only be used when a user is logged in.');
         }
 
-        $this->context->buildViolation($constraint->message)
-            ->addViolation();
+        if ($value !== $user) {
+            $this->context->buildViolation($constraint->message)
+                ->addViolation();
+        }
     }
 }
