diff --git a/tests/Functional/DragonTreasureResourceTest.php b/tests/Functional/DragonTreasureResourceTest.php
index 732cc37..846c93e 100644
--- a/tests/Functional/DragonTreasureResourceTest.php
+++ b/tests/Functional/DragonTreasureResourceTest.php
@@ -132,7 +132,7 @@ class DragonTreasureResourceTest extends ApiTestCase
                     'owner' => '/api/users/'.$user2->getId(),
                 ],
             ])
-            ->assertStatus(403)
+            ->assertStatus(422)
         ;
     }
 
