diff --git a/src/Entity/DragonTreasure.php b/src/Entity/DragonTreasure.php
index 4f4bd2e..abeac40 100644
--- a/src/Entity/DragonTreasure.php
+++ b/src/Entity/DragonTreasure.php
@@ -134,6 +134,7 @@ class DragonTreasure
     #[Groups(['treasure:read', 'treasure:write'])]
     #[Assert\Valid]
     #[IsValidOwner]
+    #[Assert\NotNull]
     #[ApiFilter(SearchFilter::class, strategy: 'exact')]
     private ?User $owner = null;
 
