diff --git a/src/ApiResource/UserApi.php b/src/ApiResource/UserApi.php
index c4c2457..44deb65 100644
--- a/src/ApiResource/UserApi.php
+++ b/src/ApiResource/UserApi.php
@@ -7,13 +7,28 @@ use ApiPlatform\Doctrine\Orm\State\Options;
 use ApiPlatform\Metadata\ApiFilter;
 use ApiPlatform\Metadata\ApiProperty;
 use ApiPlatform\Metadata\ApiResource;
+use ApiPlatform\Metadata\Delete;
+use ApiPlatform\Metadata\Get;
+use ApiPlatform\Metadata\GetCollection;
+use ApiPlatform\Metadata\Patch;
+use ApiPlatform\Metadata\Post;
 use App\Entity\DragonTreasure;
 use App\State\EntityClassDtoStateProcessor;
 use App\State\EntityToDtoStateProvider;
 use App\Entity\User;
+use Symfony\Component\Validator\Constraints as Assert;
 
 #[ApiResource(
     shortName: 'User',
+    operations: [
+        new Get(),
+        new GetCollection(),
+        new Post(
+            validationContext: ['groups' => ['Default', 'postValidation']],
+        ),
+        new Patch(),
+        new Delete(),
+    ],
     paginationItemsPerPage: 5,
     provider: EntityToDtoStateProvider::class,
     processor: EntityClassDtoStateProcessor::class,
@@ -27,14 +42,18 @@ class UserApi
     #[ApiProperty(readable: false, writable: false, identifier: true)]
     public ?int $id = null;
 
+    #[Assert\NotBlank]
+    #[Assert\Email]
     public ?string $email = null;
 
+    #[Assert\NotBlank]
     public ?string $username = null;
 
     /**
      * The plaintext password when being set or changed.
      */
     #[ApiProperty(readable: false)]
+    #[Assert\NotBlank(groups: ['postValidation'])]
     public ?string $password = null;
 
     /**
