diff --git a/src/Validator/IsValidOwnerValidator.php b/src/Validator/IsValidOwnerValidator.php
index 8b409c0..a08023f 100644
--- a/src/Validator/IsValidOwnerValidator.php
+++ b/src/Validator/IsValidOwnerValidator.php
@@ -29,6 +29,10 @@ class IsValidOwnerValidator extends ConstraintValidator
             throw new \LogicException('IsOwnerValidator should only be used when a user is logged in.');
         }
 
+        if ($this->security->isGranted('ROLE_ADMIN')) {
+            return;
+        }
+
         if ($value !== $user) {
             $this->context->buildViolation($constraint->message)
                 ->addViolation();
