diff --git a/src/Security/Voter/DragonTreasureVoter.php b/src/Security/Voter/DragonTreasureVoter.php
index ef95d1f..9c5254d 100644
--- a/src/Security/Voter/DragonTreasureVoter.php
+++ b/src/Security/Voter/DragonTreasureVoter.php
@@ -30,6 +30,10 @@ class DragonTreasureVoter extends Voter
             return false;
         }
 
+        if ($this->security->isGranted('ROLE_ADMIN')) {
+            return true;
+        }
+
         assert($subject instanceof DragonTreasure);
 
         // ... (check conditions and return true to grant permission) ...
