diff --git a/src/Security/Voter/DragonTreasureVoter.php b/src/Security/Voter/DragonTreasureVoter.php
index 64c6cad..ef95d1f 100644
--- a/src/Security/Voter/DragonTreasureVoter.php
+++ b/src/Security/Voter/DragonTreasureVoter.php
@@ -3,6 +3,7 @@
 namespace App\Security\Voter;
 
 use App\Entity\DragonTreasure;
+use Symfony\Bundle\SecurityBundle\Security;
 use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
 use Symfony\Component\Security\Core\Authorization\Voter\Voter;
 use Symfony\Component\Security\Core\User\UserInterface;
@@ -11,6 +12,10 @@ class DragonTreasureVoter extends Voter
 {
     public const EDIT = 'EDIT';
 
+    public function __construct(private Security $security)
+    {
+    }
+
     protected function supports(string $attribute, mixed $subject): bool
     {
         return in_array($attribute, [self::EDIT])
@@ -30,6 +35,10 @@ class DragonTreasureVoter extends Voter
         // ... (check conditions and return true to grant permission) ...
         switch ($attribute) {
             case self::EDIT:
+                if (!$this->security->isGranted('ROLE_TREASURE_EDIT')) {
+                    return false;
+                }
+
                 if ($subject->getOwner() === $user) {
                     return true;
                 }
