diff --git a/src/Factory/UserFactory.php b/src/Factory/UserFactory.php
index 850f064..3b4c043 100644
--- a/src/Factory/UserFactory.php
+++ b/src/Factory/UserFactory.php
@@ -52,6 +52,11 @@ final class UserFactory extends ModelFactory
         parent::__construct();
     }
 
+    public function withRoles(array $roles): self
+    {
+        return $this->addState(['roles' => $roles]);
+    }
+
     /**
      * @see https://symfony.com/bundles/ZenstruckFoundryBundle/current/index.html#model-factories
      *
diff --git a/tests/Functional/DragonTreasureResourceTest.php b/tests/Functional/DragonTreasureResourceTest.php
index 9ddb631..f38d5ef 100644
--- a/tests/Functional/DragonTreasureResourceTest.php
+++ b/tests/Functional/DragonTreasureResourceTest.php
@@ -138,6 +138,6 @@ class DragonTreasureResourceTest extends ApiTestCase
 
     public function testAdminCanPatchToEditTreasure(): void
     {
-        $admin = UserFactory::createOne(['roles' => ['ROLE_ADMIN']]);
+        $admin = UserFactory::new()->withRoles(['ROLE_ADMIN'])->create();
     }
 }
