diff --git a/tests/Functional/DragonTreasureResourceTest.php b/tests/Functional/DragonTreasureResourceTest.php
index 22bc6fd..0711688 100644
--- a/tests/Functional/DragonTreasureResourceTest.php
+++ b/tests/Functional/DragonTreasureResourceTest.php
@@ -135,4 +135,9 @@ class DragonTreasureResourceTest extends ApiTestCase
             ->assertStatus(403)
         ;
     }
+
+    public function testAdminCanPatchToEditTreasure(): void
+    {
+        $admin = UserFactory::createOne(['roles' => ['ROLE_ADMIN']]);
+    }
 }
