diff --git a/tests/Functional/DragonTreasureResourceTest.php b/tests/Functional/DragonTreasureResourceTest.php
index 698009e..b0b05f0 100644
--- a/tests/Functional/DragonTreasureResourceTest.php
+++ b/tests/Functional/DragonTreasureResourceTest.php
@@ -118,4 +118,9 @@ class DragonTreasureResourceTest extends ApiTestCase
             ->assertStatus(403)
         ;
     }
+
+    public function testAdminCanPatchToEditTreasure(): void
+    {
+        $admin = UserFactory::createOne(['roles' => ['ROLE_ADMIN']]);
+    }
 }
